<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Affymetrix Power Tools: MANUAL: apt-genotype-axiom (2.4.0)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="LICENSE.html"><span>License</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>MANUAL: apt-genotype-axiom (2.4.0) </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction.</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#options">Options</a></li>
<li><a href="#references">References</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p><b>apt-genotype-axiom</b> is a program for making genotyping calls on the Affymetrix Axiom&trade; family of arrays. <code>apt-genotype-axiom</code> uses the BRLMM-P algorithm for genotype calling. BRLMM-P is a model-based approach which performs 1-dimensional clustering by fitting a Gaussian mixture model. (Users wishing to use other legacy genotyping algorithms are recommended to use the <code>apt-probeset-genotype</code> application.)</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2>Quick Start </h2>
<p>The basic requirements for a run of apt-probeset-genotype are:</p>
<ul>
<li>A collection of CEL files to process. Note that the program will run <em>much faster</em> if the CEL files are located on a local disk as opposed to being read across a network.</li>
<li>The CDF file corresponding to the array type of the CELs. CDF files can be downloaded from <a href="http://www.affymetrix.com/support/technical/libraryfilesmain.affx">http://www.affymetrix.com/support/technical/libraryfilesmain.affx</a></li>
</ul>
<p>Additional parameters may be specified on the command line or by means of a parameter XML file.</p>
<p>The command below runs the Axiom &trade; GT1 algorithm on Axiom &trade; arrays.</p>
<pre class="fragment">  apt-genotype-axiom \
      --analysis-files-path /library/file/path \
      --arg-file Axiom_GW_Hu_SNP.r2.AxiomGT1.xml \
      --out-dir out \
      --cel-files cel_file_list.txt \
      --log-file ./apt-genotype-axiom.log
</pre><p>Alternatively, parameters can be set using command line parameters directly:</p>
<pre class="fragment">  apt-genotype-axiom \
    --analysis-files-path /library/file/path \
    --dual-channel-normalization true \
    --analysis-name AxiomGT1 \
    --chip-type Axiom_GW_Hu_SNP.r2 \
    --enable true \
    --artifact-reduction-clip 0.4 \
    --artifact-reduction-close 2 \
    --artifact-reduction-open 2 \
    --artifact-reduction-fringe 4 \
    --artifact-reduction-cc 2 \
    --sketch-target-scale-value 1000 \
    --sketch-size 50000 \
    --brlmmp-CM 1 \
    --brlmmp-bins 100 \
    --brlmmp-mix 1 \
    --brlmmp-bic 2 \
    --brlmmp-lambda 1.0 \
    --brlmmp-HARD 3 \
    --brlmmp-SB 0.75 \
    --brlmmp-transform MVA \
    --brlmmp-copyqc 0.00000 \
    --brlmmp-wobble 0.05 \
    --brlmmp-MS 0.15 \
    --brlmmp-copytype -1 \
    --brlmmp-clustertype 2 \
    --brlmmp-ocean 0.00001 \
    --brlmmp-CSepPen 0.1 \
    --brlmmp-CSepThr 4 \
    --snp-priors-input-file Axiom_GW_Hu_SNP.r2.generic_prior.txt \
    --probeset-ids Axiom_GW_Hu_SNP.r2.r2.PolyHighRez.random_20K.autosomal.ps \
    --cdf-file Axiom_GW_Hu_SNP.r2.cdf \
    --special-snps Axiom_GW_Hu_SNP.r2.specialSNPs \
    --x-probes-file Axiom_GW_Hu_SNP.r2.chrXprobes \
    --y-probes-file Axiom_GW_Hu_SNP.r2.chrYprobes \
    --sketch-target-input-file Axiom_GW_Hu_SNP.r2.AxiomGT1.sketch \
    --igender-female-threshold 1.05 \
    --igender-male-threshold 1.1 \
    --allele-summaries true \
    --snp-posteriors-output true \
    --out-dir out \
    --cel-files cel_file_list.txt \
    --log-file ./apt-genotype-axiom.log
</pre><p><a class="anchor" id="options"></a> </p>
<h2>Options: </h2>
<pre class="fragment">
   'apt-genotype-axiom'
   2.4.0 svn://svn.affymetrix.com/apt2-genotyping/branches/APT_GT_2_4_0_BRANCH@885(1494) 2015-08-14-06:11

USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    &lt;Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" /&gt;.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --out-dir (string8) = The output directory for result files.

  --temp-dir (string8) = The output directory for temporary files.

  --analysis-files-path (dirname) = Default directory to search for analysis library files

  --chp-a5-output (bool) = Output resulting calls in directory called 'cc-chp' under out-dir.
    default value: 'False'
    Aliases: --cc-chp-output

  --chp-a5-out-dir (string8) = Override the default directory for chp output.
    default value: 'cc-chp'
    Aliases: --cc-chp-out-dir

  --batch-folder (string8) = Full path to Agatha batch folder. All genotyping data will be streamed into this folder as'all_genotypes_by_snps.CHP.bin' and 'all_genotypes_by_snps.CHP.index.txt'.
    Aliases: --batch-folder

  --force (bool) = Force the files to process regardless of array type.
    default value: 'False'

  --delete-temp-files (bool) = Delete the temporary files after the run completes.
    default value: 'True'

  --chip-type (vec_string8,required) = Text identifier for chip.
    Aliases: --array-type

  --analysis-name (string8) = Name to prepend to output files.

  --report (bool) = Print report.txt file of summary and genotyping statistics.
    default value: 'True'

  --report-file (string8) = Optional full path of report.txt file.

  --summaries-only (bool) = Skip the genotyping step.
    default value: 'False'

  --x-probes-file (filename) = The file that identifies the X chromosome probe ids.

  --y-probes-file (filename) = The file that identifies the Y chromosome probe ids.

  --w-probes-file (filename) = The file that identifies the W chromosome probe ids.

  --z-probes-file (filename) = The file that identifies the Z chromosome probe ids.

  --sketch-target-use-avg (bool) = When there is a range of identical intensities, use the average of the corresponding target sketch values.
    default value: 'True'

  --probabilities-output (bool) = Output a file with comma-separated probabilities for BB,AB,AA,Ocean for each probeset and sample [default 'false']
    default value: 'False'

  --cdf-file (filename) = The CDF input file.

  --special-snps (string8) = File specifying non-autosomal chromosome information.

  --probeset-ids (string8) = Tab delimited file with column 'probeset_id' specifying probesets to genotype.

  --cel-files (filename,required) = The file that identifies the cel files to process.

  --snp-specific-param-file (filename) = File specifying SNP-specific paramaters overriding standard parameters in genotyping.
    File must have columns: ProbeSetName, ShellBarrier, Ocean, Confidence, CSepPen and CSepThr.
    Aliases: --ssp-file

  --gender-file (filename) = File specifying genders for each CEL file.
    File must have columns cel_files and gender.
    Aliases: --read-genders

  --inbred-file (filename) = File specifying inbred het penalty for each CEL file.
    File must have columns cel_files and inbred_het_penalty.
    Aliases: --read-inbred

  --hints-file (filename) = File specifying seed genotypes instead of DM algorithm
    File must have first column 'probeset_id' and one additional column per CEL file, with file basename as column name.
    Rows must have probeset id for first column and integers corresponding to calls for remaining columns.
    Calls represented by 'No Call' = -1, 'AA' = 0, 'AB' = 1, 'BB' = 2.
    Aliases: --genotypes

  --igender-female-threshold (float64) = The female threshold to use in the raw intensity gender call.
    default value: '0.480000'

  --igender-male-threshold (float64) = The male threshold to use in the raw intensity gender call.
    default value: '0.700000'

  --artifact-reduction-clip (float64) = Threshold for intensity log-ratio to be outlier
    default value: '2.000000'

  --artifact-reduction-open (int32) = Minimum size of snow (isolated residuals) to remove
    default value: '1'

  --artifact-reduction-close (int32) = Max gap to close between tentative artifacts
    default value: '3'

  --artifact-reduction-fringe (int32) = Add a fringe to catch border features that might be poor.
    default value: '2'

  --artifact-reduction-cc (int32) = How many channels must support an artifact
    default value: '1'

  --artifact-reduction-trustcheck (bool) = Completely untrusted, blemished probesets in samples are no-calls
    default value: 'False'

  --sketch-target-input-file (string8) = Name of the file containing sketch to use.
    Aliases: --target-sketch

  --sketch-target-output (bool) = Write the quantile normalization distribution (or sketch) to a file for reuse with target-sketch option. [default 'false']
    default value: 'False'
    Aliases: --write-sketch

  --sketch-target-output-file (string8) = Optional fully-specified path of sketch.txt file.

  --sketch-target-scale-value (float64) = Target to scale the sketch to
    default value: '0.000000'

  --dual-channel-target-sketch (bool) = Create sketches from separate channels (AT/GC)
    default value: 'True'

  --sketch-size (int32,required) = The sketch size to use.

  --dual-channel-normalization (bool,required) = Use dual channel normalization.

  --sample-multiplier (int32) = Used for generating fake blob with large number of samples. Ex: 10 samples x 500 multiplier = 5000 fake samples.
    default value: '1'

  --allele-summaries (bool) = Print file of intensities summarized by allele.
    default value: 'False'
    Aliases: --summaries

  --allele-summaries-file (string8) = Optional fully-specified path of allele-summaries.txt file.
    Aliases: --summaries-file

  --snp-priors-input-file (string8,required) = The string that identifies the snp priors to use when genotyping.

  --snp-posteriors-output (bool) = Write snp posteriors to output when genotyping.
    default value: 'False'
    Aliases: --write-models

  --snp-posteriors-output-file (string8) = Optional full path of snp-posteriors.txt file.

  --table-output (bool) = Output matching matrices of tab delimited genotype calls and confidences. [default 'true']
    default value: 'True'

  --calls-file (string8) = Optional full path of calls.txt file.

  --confidences-file (string8) = Optional full path of confidences.txt file.

  --probabilities-file (string8) = Optional full path of probabilities template file.  The file is typically partitioned and integers inserted into the filename before the .txt suffix.

  --probabilities-file-sample-count (int32) = Number of samples per probability file.  These files can be large, especially when running 1000's of samples. [default '0' = don't break up the file]
    default value: '0'
    Aliases: --prob-file-sample-count

  --brlmmp-HARD (int32) = brlmmp hardshell
    default value: '3'

  --brlmmp-SB (float64) = brlmmp shellbarrier
    default value: '0.450000'

  --brlmmp-CM (int32) = brlmmp callmethod (1=batch mode, 2=single sample mode)
    default value: '1'

  --brlmmp-bins (int32) = brlmmp bins
    default value: '100'

  --brlmmp-mix (int32) = brlmmp mix
    default value: '1'

  --brlmmp-bic (float64) = brlmmp bic
    default value: '2.000000'

  --brlmmp-CSepPen (float64) = brlmmp CSepPen
    default value: '0.000000'

  --brlmmp-CSepThr (float64) = brlmmp CSepThr
    default value: '16.000000'

  --brlmmp-lambda (float64) = brlmmp lambda
    default value: '1.000000'

  --brlmmp-wobble (float64) = brlmmp wobble
    default value: '0.050000'

  --brlmmp-copyqc (float64) = brlmmp copyqc. 0 = no test for copy qc
    default value: '0.000000'

  --brlmmp-copytype (int32) = brlmmp copytype. 0 = standard copy qc method
    default value: '0'

  --brlmmp-MS (float64) = brlmmp MS. confidence threshold
    default value: '0.050000'

  --brlmmp-ocean (float64) = brlmmp ocean
    default value: '0.000000'

  --brlmmp-clustertype (int32) = brlmmp clustertype. 1 = standard 1-d clustering only
    default value: '1'

  --brlmmp-hints (int32) = brlmmp hints
    default value: '0'

  --brlmmp-IsoHetY (float64) = brlmmp IsoHetY
    default value: '0.000000'

  --brlmmp-inflatePRA (float64) = brlmmp inflatePRA. 0 = no increase in uncertainty
    default value: '0.000000'

  --brlmmp-CP (float64) = brlmmp contradiction penalty. 0 = no penalty
    default value: '16.000000'

  --@help       = print developer help

</pre><p><a class="anchor" id="faq"></a> </p>
<h2>Frequently Asked Questions </h2>
<p><a class="anchor" id="references"></a> </p>
<h2>References </h2>
<ul>
<li>BRLMM-P - <a href="http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf">http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf</a></li>
<li>PLIER - <a href="http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf">http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf</a> </li>
</ul>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated for Affymetrix Power Tools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
