<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Affymetrix Power Tools: MANUAL: apt-copynumber-onco-som-ssa (2.4.0)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="LICENSE.html"><span>License</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>MANUAL: apt-copynumber-onco-som-ssa (2.4.0) </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction.</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#options">Options</a></li>
<li><a href="#references">References</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p><b>apt-copynumber-onco-som-ssa</b> implements the somatic mutation analysis pipeline for OncoScan arrays. <code>apt-copynumber-onco-som-ssa</code> is designed to be run sequentially with <code>apt-copynumber-onco-ssa</code>. Input for each OncoScan experiment consists of a pair of CEL files for AT and GC channels and an OSCHP file generated by <code>apt-copynumber-onco-ssa</code>. Rather than generating a separate output file for somatic mutation results, <code>apt-copynumber-onco-som-ssa</code> will add a <em>somatic mutation</em> data group to an existing OSCHP file. <code>apt-copynumber-onco-som-ssa</code> supports two modes of operation: single sample mode and "with matched normal" analysis. In the "with matched normal" mode a pair of cel files is identified as “normal” and a pair as “tumor” and the normal data is used to identify heterozygous markers for use by the algorithm when analyzing the tumor pair. In this analysis the log2ratios are still calculated relative to the reference model file. (This is not a “paired analysis” where the tumor is compared to the normal to calculate the log2 ratio data.) experiments (i.e. four CEL files) are consisting of a “normal” and a “tumor” sample are submitted.</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2>Quick Start </h2>
<h3>Input/Output Files</h3>
<p>The following input files are required by apt-copynumber-onco-som-ssa:</p>
<table class="doxtable">
<tr>
<th>Name </th><th>File extension/type </th><th>Description  </th></tr>
<tr>
<td>reference-file </td><td>REF_MODEL (HDF5) </td><td>Reference data generated by apt-copynumber-onco-ref </td></tr>
<tr>
<td>som-thresholds-file </td><td>text </td><td>Thresholds for Somatic Mutation calls </td></tr>
<tr>
<td>annotation-file </td><td>annot.db (SQLite) </td><td>Annotation data for probes and probesets </td></tr>
<tr>
<td>x-probes-file </td><td>chrXprobes (text) </td><td>List of X-chromosome probes </td></tr>
<tr>
<td>y-probes-file </td><td>chrYprobes (text) </td><td>List of Y-chromosome probes </td></tr>
<tr>
<td>cel-pairs-file </td><td>tsv </td><td>Cel file pairs (AT and GC channel for each sample) for experiments </td></tr>
</table>
<p>The following output files are updated by apt-copynumber-onco:</p>
<table class="doxtable">
<tr>
<th>Name </th><th>File extension/type </th><th>Description  </th></tr>
<tr>
<td>OSCHP file </td><td>OSCHP (HDF5) </td><td>Copynumber and other analysis data for each experiment </td></tr>
</table>
<h3>CEL files lists:</h3>
<p>The CEL file pairs are specified in a tab delimited text file. In single-sample mode this file will have either two or three columns, with two columns representing the AT-channel and the GC-channel CEL files. In normal/tumor pair mode, the file will have either four or five columns, with four columns representing the normal AT and GC-channel CEL files, and the tumor AT and GC-channel CEL files. The optional third or fifth column can be used to specify the name of the output OSCHP files. (If omitted an OSCHP file name is generated based on the CEL file names.)</p>
<p>The first row of the cel-pairs-file should contain the column names. Supported column names are:</p>
<pre class="fragment">        atchannelcel    - the AT channel CEL file
        gcchannelcel    - the GC channel CEL file
        atchannelmatchednormalcel       - the AT channel CEL file for matched normal sample
        gcchannelmatchednormalcel       - the GC channel CEL file for matched normal sample
        result          - the OSCHP file name to be generated (optional)
</pre><p>See <a href="VIGNETTE-OncoScan-ssa.html">OncoScan single-sample analysis</a> for examples.</p>
<p><a class="anchor" id="options"></a> </p>
<h2>Options: </h2>
<pre class="fragment">
   'apt-copynumber-onco-som-ssa'
   2.4.0 svn://svn.affymetrix.com/apt2-copynumber/branches/APT_CN_2_4_0_BRANCH@1018(1495) 2015-08-25-04:44

USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    &lt;Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" /&gt;.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --out-dir (string8) = The output directory for result files.

  --temp-dir (string8) = The output directory for temporary files.

  --force (bool) = Force the files to process regardless of array type.
    default value: 'False'

  --keep-intermediate-data (bool) = Dump data on itermediate calculations for testing purposes.
    default value: 'False'

  --keep-combined-cels (bool) = Keep the combined cel file data.
    default value: 'False'

  --reference-file (string8,required) = The reference input file.

  --set-analysis-name (string8) = Analysis name to use as prefix for output files.

  --file-name-prefix (string8) = OSCHP file name prefix

  --file-name-suffix (string8) = OSCHP file name suffix

  --annotation-file (string8,required) = The annotation input file. Must have the Annotations_SM table.

  --x-probes-file (string8) = The file that identifies the X chromosome probe ids.

  --y-probes-file (string8) = The file that identifies the Y chromosome probe ids.

  --cel-pairs-file (string8) = The file that identifies the cel file pairs to process.

  --cel-files (string8) = The file that identifies the cel files to process.

  --loh-files (string8) = The file that identifies the loh files to process.

  --som-thresholds-file (string8) = The string that identifies the Somatic mutation probe set possible and present thresholds.

  --agr-denominator-source (string8) = The denominator to use in calculating the antigenomic ratio.
    default value: 'SNP'

  --agr-denominator-percentile (float64) = The percentile denominator to use in calculating the antigenomic ratio denominator.
    default value: '0.750000'

  --igender-female-threshold (float64) = The female threshold to use in the raw intensity gender call.
    default value: '0.440000'

  --igender-male-threshold (float64) = The male threshold to use in the raw intensity gender call.
    default value: '0.630000'

  --nddicov-min-nd-probe-count-per-bin (int32) = Override the icov-min-nd-probe-count-per-bin parameter. This is intended to be used in SSA mode.
    default value: '30'

  --nddicov-use-nd-signal-markers (string8) = Override the icov-use-nd-signal-markers parameter (on/off). This is intended to be used in SSA mode.
    default value: 'off'

  --dual-channel-normalization (bool) = Use dual channel normalization.
    default value: 'True'

  --norm-use-autosomal (bool) = Use autosomal probe sets in the sample sketch normalization.
    default value: 'False'

  --norm-use-autosomal-nd (bool) = Use normal diploid autosomal probe sets in the sample sketch normalization.
    default value: 'True'

  --norm-use-som (bool) = Use somatic mutation probe sets in the sample sketch normalization.
    default value: 'False'

  --norm-sketch-probesets-file (filename) = Probe set file listing probe sets that are to be used in the sample sketch normalization.

  --meta-data-info (vec_string8) = Meta data in key=value pair format that will be written to the header.

  --num-out-med (int32) = Number of outliers to remove from median calculation in adjusted SM score.
    default value: '0'

  --num-out-std (int32) = Number of outliers to remove from standard deviation calculation in adjusted SM score.
    default value: '2'

  --adj-mut-score (bool) = Adjusted Somatic Mutation scores on or off
    default value: 'False'

  --min-mut-score (float64) = Minimum Median Somatic Mutation score
    default value: '0.000000'

  --@help       = print developer help

</pre><p><a class="anchor" id="faq"></a> </p>
<h2>Frequently Asked Questions </h2>
<p><a class="anchor" id="references"></a> </p>
<h2>References </h2>
<ul>
<li>BRLMM-P - <a href="http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf">http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf</a></li>
<li>PLIER - <a href="http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf">http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf</a> </li>
</ul>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated for Affymetrix Power Tools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
