<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Affymetrix Power Tools: MANUAL: apt-copynumber-cyto-ssa (2.4.0)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="LICENSE.html"><span>License</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>MANUAL: apt-copynumber-cyto-ssa (2.4.0) </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction.</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#options">Options</a></li>
<li><a href="#references">References</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p><b>apt-copynumber-cyto-ssa</b> implements the copynumber and loss of heterozygosity (LOH) analysis pipeline for the CytoScan family of arrays on a a per sample basis with respect to a reference set of samples. The copy number algorithm it implements assumes that the reference set comprises a mix of normal human males (with XY chromosomes) and normal human females (with XX chromosomes). The algorithms assume that in this reference for each autosomal marker (SNP or Copy Number probe) the predominant Copy Number is 2, and for the sex chromosomes the copy number is determined by the gender.</p>
<p><code>apt-copynumber-cyto-ssa</code> supports two modes of operation: with or without <b>normal diploid</b> (ND) normalization. The normal mode of use (without ND normalization) assumes that most of the genome of the sample is diploid. This is generally a safe assumption for constitutive samples, but sometimes becomes invalid for other sample types. When the expectation is that the sample is not mostly diploid, the ND pipeline is recommended.</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2>Quick Start </h2>
<h3>Input/Output Files</h3>
<p>The following input files are required by apt-copynumber-cyto-ssa:</p>
<table class="doxtable">
<tr>
<th>Name </th><th>File extension/type </th><th>Description  </th></tr>
<tr>
<td>reference-file </td><td>REF_MODEL (HDF5) </td><td>Reference data generated by apt-copynumber-cyto-ref </td></tr>
<tr>
<td>snp-qc-snp-list </td><td>snplisttxt </td><td>SNP probes used in calculating processed SNPQC and Raw SNPQC. The first row of the file must be "probeset_id". </td></tr>
<tr>
<td>annotation-file </td><td>annot.db (SQLite) </td><td>Annotation data for probes and probesets </td></tr>
<tr>
<td>x-probes-file </td><td>chrXprobes (text) </td><td>List of X-chromosome probes. Should have a single column with label <code>probe_id</code>. Used for copy number probe chrX/Y ratio gender calling. </td></tr>
<tr>
<td>y-probes-file </td><td>chrYprobes (text) </td><td>List of Y-chromosome probes and probesets. Should have first column with label <code>probe_id</code> and second with label <code>probeset_id</code>. Probe_id is used for copy number probe chrX/Y ratio gender calling and probeset_id is used for hasY gender call and Y-Gender call. </td></tr>
<tr>
<td>cel-files </td><td>text </td><td>A list of Cel files to be processed, one per line with first row <code>cel_files</code>. </td></tr>
</table>
<p>The following output files are generated by apt-copynumber-cyto:</p>
<table class="doxtable">
<tr>
<th>Name </th><th>File extension/type </th><th>Description  </th></tr>
<tr>
<td>CYCHP file </td><td>cychp </td><td>Copynumber and other analysis data for each experiment </td></tr>
<tr>
<td>report-file </td><td>text </td><td>Summary statistics for the experiments </td></tr>
</table>
<h3>Command-line Parameters</h3>
<p>On Unix systems, the command line to execute CytoScanHD single-sample analysis without ND normal diploid normalization would look like this:</p>
<pre class="fragment">apt-copynumber-cyto-ssa \
    --allele-peak-count 24 \
    --nearest-power-of-two 512 \
    --keep-intermediate-data false \
    --dual-channel-normalization false \
    --mangle-probeset-names false \
    --waviness-block-size 50 \
    --waviness-genomic-span 0 \
    --igender-male-threshold 1.5 \
    --igender-female-threshold 0.9 \
    --sig-gender-xx-cutoff 0.61 \
    --sig-gender-xx-cutoff-high 0.95 \
    --sig-gender-y-cutoff 0.58 \
    --sig-gender-reference-chromosome 2 \
    --force false \
    --text-output false \
    --oschp-output false \
    --cychp-output true \
    --set-analysis-name cyhd \
    --snp-qc-snp-list $LIBDIR/CytoScanHD_Array.snplist.txt \
    --x-probes-file $LIBDIR/CytoScanHD_Array.chrXprobes \
    --y-probes-file $LIBDIR/CytoScanHD_Array.chrYprobes \
    --annotation-file $LIBDIR/CytoScanHD_Array.na33.annot.db \
    --reference-file $LIBDIR/CytoScanHD_Array.na33.r2.REF_MODEL \
    --y-target -0.50 \
    --out-dir $OUTDIR \
    --log-file $OUTDIR/apt-copynumber-cyhd-ssa.log \
    --temp-dir $OUTDIR/temp \
    --alpha-cn-calibrate 0.564278 \
    --alpha-X-cn-calibrate 0.619453 \
    --alpha-Y-cn-calibrate 0.494620 \
    --beta-cn-calibrate 1 \
    --beta-X-cn-calibrate 1 \
    --beta-Y-cn-calibrate 1 \
    --brlmmp-HARD 3 \
    --brlmmp-SB 0.450000 \
    --brlmmp-CM 2 \
    --brlmmp-bins 100 \
    --brlmmp-mix 1 \
    --brlmmp-bic 2 \
    --brlmmp-CSepPen 0.0 \
    --brlmmp-CSepThr 16 \
    --brlmmp-lambda 1.0 \
    --brlmmp-wobble 0.05 \
    --brlmmp-copyqc 0.000000 \
    --brlmmp-copytype 0 \
    --brlmmp-ocean 0.0 \
    --brlmmp-clustertype 1 \
    --brlmmp-transform mva \
    --brlmmp-MS 0.05 \
    --log2-ratio-adjuster-image-node:enable=true \
    --hpf-mini-block-rows 8 \
    --hpf-mini-block-cols 8 \
    --hpf-global-smooth-weight 256 \
    --hpf-local-smooth-weight 64 \
    --hpf-converged 0.000100 \
    --log2-ratio-adjuster-wave-node:enable=true \
    --wave-bandwidth 101 \
    --wave-bin-count 25 \
    --wave-count 6 \
    --wave-smooth true \
    --median-autosome-median-normalization true \
    --ad-outlier-trim 3.0 \
    --ap-ad-step 20 \
    --ap-ad-window 100 \
    --ap-ad-point-count 129 \
    --ap-ad-bandwidth 0.25 \
    --ap-ad-cutoff 0.05 \
    --ap-ad-threshold 0.35 \
    --ap-ad-symmetry true \
    --ap-ad-height-threshold 0.000000 \
    --ap-ad-height-threshold-bound 1024.000000 \
    --normal-diploid-normalization false \
    --ndd-weights-type equal \
    --ndd-point-count 128 \
    --ndd-bandwidth 0.25 \
    --ndd-cutoff 0.05 \
    --ndd-step 40 \
    --ndd-window 275 \
    --ndd-min-markers 10 \
    --ndd-log2-threshold 0.28 \
    --ndd-th1 0,0.4 \
    --ndd-th2 0.7,1.3 \
    --ndd-th3 0.7,1.3 \
    --ndd-th4 0,0.4 \
    --ndd-min-percent-nd 0 \
    --ndd-min-number-nd 8000 \
    --ndd-max-number-nd 1500000 \
    --ndd-med-l2r-low-percentile-ND 0.015 \
    --ndd-med-l2r-threshold-ND 0.20 \
    --ndd-iqr-factor-coarse 5.0 \
    --ndd-iqr-factor-fine 3.0 \
    --nddlrcov-min-nd-marker-count-per-bin 30 \
    --nddlrcov-use-nd-lr-markers on \
    --nddqc-min-count 2000 \
    --nddicov-min-nd-probe-count-per-bin 30 \
    --nddicov-use-nd-signal-markers off \
    --kernel-sigma-span 50 \
    --hmmCN_state 0,1,2,3,4 \
    --hmmCN_mu -2,-0.5,0,0.33,0.54 \
    --hmmCN_sigma 0.3,0.3,0.3,0.3,0.3 \
    --hmmCN_state-X 0,1,2,3,4 \
    --hmmCN_mu-X -2,-0.55,0,0.36,0.57 \
    --hmmCN_sigma-X 0.3,0.3,0.3,0.3,0.3 \
    --hmmCN_state-Y 0,1,2,3,4 \
    --hmmCN_mu-Y -1.4,-0.5,0,0.3,0.51 \
    --hmmCN_sigma-Y 0.3,0.3,0.3,0.3,0.3 \
    --diagonal-weight-Y 0.995 \
    --mapd-weight-Y 0.2 \
    --min-segment-size-Y 5 \
    --hmm-confidence-weight-Y 0.6 \
    --diagonal-weight 0.995 \
    --mapd-weight 0.2 \
    --min-segment-size 5 \
    --hmm-confidence-weight 0.6 \
    --hmm-shrink true \
    --diagonal-weight-X 0.995 \
    --mapd-weight-X 0.2 \
    --min-segment-size-X 5 \
    --hmm-confidence-weight-X 0.6 \
    --agr-denominator-source CN \
    --agr-denominator-percentile 0.5 \
    --cn-gender-cutoff 0.5 \
    --loh-error-rate 0.05 \
    --loh-beta 0.001 \
    --loh-alpha 0.01 \
    --loh-separation 1000000 \
    --loh-min-marker-count 10 \
    --loh-no-call-threshold 0.05 \
    --loh-min-genomic-span 1000000 \
    --gains-boundries 0.0820,0.1588,0.2318,0.3130 \
    --losses-boundries -0.1049,-0.2263,-0.3477,-0.4691 \
    --marker-bandwidth 6000 \
    --confidence-window 251 \
    --run-y-chromosome false \
    --cel-files $CELFileList \
    --cn-neutral-loh-node:enable=true
</pre><p>Here the environment variable <code>$LIBDIR</code> is set to the location of the library files, and the environment variable <code>$OUTDIR</code> is set to the location to place the output files.</p>
<p>WARNING: apt-copynumber-cyto-ssa will overwrite any existing output files it finds. If you wish to keep existing results make sure to specify a different output directory name.</p>
<p>NOTE: The windows DOS prompt also does not allow a continuation of a command with the '\' character, unlike unix. So in the examples shown here the '\' character should be omitted and everything entered on a single line.</p>
<p>On Unix systems, the command line to execute CytoScanHD single-sample analysis with ND normal diploid normalization would look like this:</p>
<pre class="fragment">apt-copynumber-cyto-ssa \
    --allele-peak-count 24 \
    --nearest-power-of-two 512 \
    --keep-intermediate-data false \
    --dual-channel-normalization false \
    --mangle-probeset-names false \
    --waviness-block-size 50 \
    --waviness-genomic-span 0 \
    --igender-male-threshold 1.5 \
    --igender-female-threshold 0.9 \
    --sig-gender-xx-cutoff 0.61 \
    --sig-gender-xx-cutoff-high 0.95 \
    --sig-gender-y-cutoff 0.58 \
    --sig-gender-reference-chromosome 2 \
    --force false \
    --text-output false \
    --oschp-output false \
    --cychp-output true \
    --set-analysis-name cyhd \
    --snp-qc-snp-list $LIBDIR/CytoScanHD_Array.snplist.txt \
    --x-probes-file $LIBDIR/CytoScanHD_Array.chrXprobes \
    --y-probes-file $LIBDIR/CytoScanHD_Array.chrYprobes \
    --annotation-file $LIBDIR/CytoScanHD_Array.na33.annot.db \
    --reference-file $LIBDIR/CytoScanHD_Array.na33.r2.REF_MODEL \
    --y-target -0.50 \
    --out-dir $OUTDIR \
    --log-file $OUTDIR/apt-copynumber-cyhd-ssa.log \
    --temp-dir $OUTDIR/temp \
    --alpha-cn-calibrate 0.564278 \
    --alpha-X-cn-calibrate 0.619453 \
    --alpha-Y-cn-calibrate 0.494620 \
    --beta-cn-calibrate 1 \
    --beta-X-cn-calibrate 1 \
    --beta-Y-cn-calibrate 1 \
    --brlmmp-HARD 3 \
    --brlmmp-SB 0.450000 \
    --brlmmp-CM 2 \
    --brlmmp-bins 100 \
    --brlmmp-mix 1 \
    --brlmmp-bic 2 \
    --brlmmp-CSepPen 0.0 \
    --brlmmp-CSepThr 16 \
    --brlmmp-lambda 1.0 \
    --brlmmp-wobble 0.05 \
    --brlmmp-copyqc 0.000000 \
    --brlmmp-copytype 0 \
    --brlmmp-ocean 0.0 \
    --brlmmp-clustertype 1 \
    --brlmmp-transform mva \
    --brlmmp-MS 0.05 \
    --log2-ratio-adjuster-image-node:enable=false \
    --hpf-mini-block-rows 8 \
    --hpf-mini-block-cols 8 \
    --hpf-global-smooth-weight 256 \
    --hpf-local-smooth-weight 64 \
    --hpf-converged 0.000100 \
    --log2-ratio-adjuster-wave-node:enable=true \
    --wave-bandwidth 101 \
    --wave-bin-count 25 \
    --wave-count 6 \
    --wave-smooth true \
    --median-autosome-median-normalization true \
    --ad-outlier-trim 3.0 \
    --ap-ad-step 20 \
    --ap-ad-window 100 \
    --ap-ad-point-count 129 \
    --ap-ad-bandwidth 0.25 \
    --ap-ad-cutoff 0.05 \
    --ap-ad-threshold 0.35 \
    --ap-ad-symmetry true \
    --ap-ad-height-threshold 0.000000 \
    --ap-ad-height-threshold-bound 1024.000000 \
    --normal-diploid-normalization true \
    --fld-file $LIBDIR/CytoScanHD_Array.NDN.FLD.r1.txt \
    --ndd-weights-type FLD \
    --ndd-point-count 128 \
    --ndd-bandwidth 0.25 \
    --ndd-cutoff 0.05 \
    --ndd-step 40 \
    --ndd-window 275 \
    --ndd-min-markers 10 \
    --ndd-log2-threshold 0.28 \
    --ndd-th1 0,0.4 \
    --ndd-th2 0.7,1.3 \
    --ndd-th3 0.7,1.3 \
    --ndd-th4 0,0.4 \
    --ndd-min-percent-nd 0 \
    --ndd-min-number-nd 8000 \
    --ndd-max-number-nd 1500000 \
    --ndd-med-l2r-low-percentile-ND 0.015 \
    --ndd-med-l2r-threshold-ND 0.20 \
    --ndd-iqr-factor-coarse 5.0 \
    --ndd-iqr-factor-fine 3.0 \
    --nddlrcov-min-nd-marker-count-per-bin 30 \
    --nddlrcov-use-nd-lr-markers on \
    --nddqc-min-count 2000 \
    --nddicov-min-nd-probe-count-per-bin 30 \
    --nddicov-use-nd-signal-markers off \
    --kernel-sigma-span 50 \
    --hmmCN_state 0,1,2,3,4 \
    --hmmCN_mu -2,-0.5,0,0.33,0.54 \
    --hmmCN_sigma 0.3,0.3,0.3,0.3,0.3 \
    --hmmCN_state-X 0,1,2,3,4 \
    --hmmCN_mu-X -2,-0.55,0,0.36,0.57 \
    --hmmCN_sigma-X 0.3,0.3,0.3,0.3,0.3 \
    --hmmCN_state-Y 0,1,2,3,4 \
    --hmmCN_mu-Y -1.4,-0.5,0,0.3,0.51 \
    --hmmCN_sigma-Y 0.3,0.3,0.3,0.3,0.3 \
    --diagonal-weight-Y 0.995 \
    --mapd-weight-Y 0.2 \
    --min-segment-size-Y 5 \
    --hmm-confidence-weight-Y 0.6 \
    --diagonal-weight 0.995 \
    --mapd-weight 0.2 \
    --min-segment-size 5 \
    --hmm-confidence-weight 0.6 \
    --hmm-shrink true \
    --diagonal-weight-X 0.995 \
    --mapd-weight-X 0.2 \
    --min-segment-size-X 5 \
    --hmm-confidence-weight-X 0.6 \
    --agr-denominator-source CN \
    --agr-denominator-percentile 0.5 \
    --cn-gender-cutoff 0.5 \
    --loh-error-rate 0.05 \
    --loh-beta 0.001 \
    --loh-alpha 0.01 \
    --loh-separation 1000000 \
    --loh-min-marker-count 10 \
    --loh-no-call-threshold 0.05 \
    --loh-min-genomic-span 1000000 \
    --gains-boundries 0.0820,0.1588,0.2318,0.3130 \
    --losses-boundries -0.1049,-0.2263,-0.3477,-0.4691 \
    --marker-bandwidth 6000 \
    --confidence-window 251 \
    --run-y-chromosome false \
    --cel-files $CELFileList \
    --cn-neutral-loh-node:enable=true
</pre><p>NOTE: When setting normal-diploid normalization to true it is necessary to set <code>--log2-ratio-adjuster-image-node:enable</code> to <code>false</code>.</p>
<p><a class="anchor" id="options"></a> </p>
<h2>Options: </h2>
<pre class="fragment">
   'apt-copynumber-cyto-ssa'
   2.4.0 svn://svn.affymetrix.com/apt2-copynumber/branches/APT_CN_2_4_0_BRANCH@1023(1497) 2015-08-28-09:45

USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    &lt;Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" /&gt;.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --out-dir (string8) = The output directory for result files.

  --temp-dir (string8) = The output directory for temporary files.

  --force (bool) = Force the files to process regardless of array type.
    default value: 'False'

  --keep-intermediate-data (bool) = Dump data on itermediate calculations for testing purposes.
    default value: 'False'

  --snp-qc-snp-list (filename) = SNPs to use in calculating SNP QC.

  --reference-file (string8) = The reference input file.

  --y-target (float64) = The Y chromosome target log2 ratio value.
    default value: '-0.500000'

  --nddqc-min-count (int32) = Minimum number of normal diploid markers that must be found to be used as the basis for QC calculations.
    default value: '500'

  --set-analysis-name (string8) = Analysis name to use as prefix for output files.

  --file-name-prefix (string8) = OSCHP file name prefix

  --file-name-suffix (string8) = OSCHP file name suffix

  --brlmmp-HARD (int32) = brlmmp hardshell
    default value: '3'

  --brlmmp-SB (float64) = brlmmp shellbarrier
    default value: '0.450000'

  --brlmmp-CM (int32) = brlmmp callmethod (1=batch mode, 2=single sample mode)
    default value: '2'

  --brlmmp-bins (int32) = brlmmp bins
    default value: '100'

  --brlmmp-mix (int32) = brlmmp mix
    default value: '1'

  --brlmmp-bic (float64) = brlmmp bic
    default value: '2.000000'

  --brlmmp-CSepPen (float64) = brlmmp CSepPen
    default value: '0.000000'

  --brlmmp-CSepThr (float64) = brlmmp CSepThr
    default value: '16.000000'

  --brlmmp-lambda (float64) = brlmmp lambda
    default value: '1.000000'

  --brlmmp-wobble (float64) = brlmmp wobble
    default value: '0.050000'

  --brlmmp-copyqc (float64) = brlmmp copyqc. 0 = no test for copy qc
    default value: '0.000000'

  --brlmmp-copytype (int32) = brlmmp copytype. 0 = standard copy qc method
    default value: '0'

  --brlmmp-ocean (float64) = brlmmp ocean
    default value: '0.000000'

  --brlmmp-clustertype (int32) = brlmmp clustertype. 1 = standard 1-d clustering only
    default value: '1'

  --brlmmp-transform (string8) = brlmmp transform. (mva, rvt, ssf, ces, assf, ccs)
    default value: 'mva'

  --brlmmp-MS (float64) = brlmmp MS. confidence threshold
    default value: '0.050000'

  --brlmmp-hints (int32) = brlmmp hints
    default value: '0'

  --brlmmp-IsoHetY (float64) = brlmmp IsoHetY
    default value: '0.000000'

  --brlmmp-inflatePRA (float64) = brlmmp inflatePRA. 0 = no increase in uncertainty
    default value: '0.000000'

  --brlmmp-CP (float64) = brlmmp contradiction penalty. 0 = no penalty
    default value: '0.000000'

  --normal-diploid-normalization (bool) = Normal diploid processing turned off when false.
    default value: 'True'

  --alpha-cn-calibrate (float64) = IDK
    default value: '0.564278'

  --alpha-X-cn-calibrate (float64) = IDK
    default value: '0.619453'

  --alpha-Y-cn-calibrate (float64) = IDK
    default value: '0.494620'

  --beta-cn-calibrate (float64) = IDK
    default value: '1.000000'

  --beta-X-cn-calibrate (float64) = IDK
    default value: '1.000000'

  --beta-Y-cn-calibrate (float64) = IDK
    default value: '1.000000'

  --allele-peak-count (int32) = 
    default value: '24'

  --nearest-power-of-two (int32) = 
    default value: '512'

  --annotation-file (filename,required) = The annotation input file.

  --x-probes-file (filename) = The file that identifies the X chromosome probe ids.

  --y-probes-file (filename) = The file that identifies the Y chromosome probe ids.

  --cel-files (filename) = The file that identifies the cel files to process.

  --loh-files (filename) = The file that identifies the loh files to process.

  --fld-file (filename) = The string that identifies the pFLD values to load into the reference.

  --ap-baf-fld-weights (filename) = Name of file with FLD-derived weights.
    Aliases: --baf-fld-weights

  --agr-denominator-source (string8) = The denominator to use in calculating the antigenomic ratio.
    default value: 'CN'

  --agr-denominator-percentile (float64) = The percentile denominator to use in calculating the antigenomic ratio denominator.
    default value: '0.500000'

  --igender-female-threshold (float64) = The female threshold to use in the raw intensity gender call.
    default value: '0.440000'

  --igender-male-threshold (float64) = The male threshold to use in the raw intensity gender call.
    default value: '0.630000'

  --nddicov-min-nd-probe-count-per-bin (int32) = Override the icov-min-nd-probe-count-per-bin parameter. This is intended to be used in SSA mode.
    default value: '30'

  --nddicov-use-nd-signal-markers (string8) = Override the icov-use-nd-signal-markers parameter (on/off). This is intended to be used in SSA mode.
    default value: 'off'

  --dual-channel-normalization (bool) = Use dual channel normalization.
    default value: 'False'

  --ad-outlier-trim (float64) = The trim value to use for alleleic difference outliers.
    default value: '3.000000'

  --ndd-weights-type (string8) = Normal diploid detection weigths types (FLD, replicates, equal).
    default value: 'FLD'

  --ndd-point-count (int32) = Normal diploid detection density point count.
    default value: '128'

  --ndd-bandwidth (float64) = Normal diploid detection density bandwidth.
    default value: '0.250000'

  --ndd-cutoff (float64) = Normal diploid detection density cutoff.
    default value: '0.050000'

  --ndd-step (int32) = Normal diploid detection moving window step.
    default value: '40'

  --ndd-window (int32) = Normal diploid detection moving window window size.
    default value: '350'

  --ndd-min-markers (int32) = Normal diploid detection moving window min markers.
    default value: '10'

  --ndd-log2-threshold (float64) = Normal diploid detection window median log2 ratio threshold.
    default value: '0.250000'

  --ndd-th1 (string8) = Normal diploid detection th1 (pair of values).
    default value: '0,0.45'

  --ndd-th2 (string8) = Normal diploid detection th2 (pair of values).
    default value: '0.85,1.25'

  --ndd-th3 (string8) = Normal diploid detection th3 (pair of values).
    default value: '0.7,1.3'

  --ndd-th4 (string8) = Normal diploid detection th4 (pair of values).
    default value: '0,0.3'

  --ndd-min-percent-nd (float64) = Minimum percent of normal-diploid autosomal probe sets required.
    default value: '0.000000'

  --ndd-min-number-nd (int32) = Minimum number of normal-diploid autosomal probe sets required.
    default value: '30'

  --ndd-max-number-nd (int32) = Maximum number of normal-diploid autosomal probe sets required.
    default value: '100'

  --ndd-med-l2r-low-percentile-ND (float64) = Normal diploid detection median log2 ratio low percentile.
    default value: '0.005000'

  --ndd-med-l2r-threshold-ND (float64) = Normal diploid detection median log2 ratio threshold.
    default value: '0.280000'

  --ndd-iqr-factor-coarse (float64) = Normal diploid detection IQR factor (coarse).
    default value: '5.000000'

  --ndd-iqr-factor-fine (float64) = Normal diploid detection IQR factor (fine).
    default value: '3.000000'

  --sig-gender-reference-chromosome (int32) = XY signal gender reference chromosome
    default value: '2'

  --sig-gender-xx-cutoff (float64) = XY signal gender XX cutoff
    default value: '0.850000'

  --sig-gender-xx-cutoff-high (float64) = XY signal gender XX cutoff high
    default value: '1.220000'

  --sig-gender-y-cutoff (float64) = XY signal gender Y cutoff
    default value: '0.310000'

  --hpf-mini-block-rows (int32) = Mini block rows.
    default value: '8'

  --hpf-mini-block-cols (int32) = Mini block cols.
    default value: '8'

  --hpf-global-smooth-weight (float64) = Global smooth weight.
    default value: '256.000000'

  --hpf-local-smooth-weight (float64) = Local smooth weight.
    default value: '64.000000'

  --hpf-converged (float64) = Converged.
    default value: '0.000100'

  --nddlrcov-min-nd-marker-count-per-bin (int32) = Override the lrcov-min-nd-marker-count-per-bin parameter. This is intended to be used in SSA mode.
    default value: '30'

  --nddlrcov-use-nd-lr-markers (string8) = Override the lrcov-use-nd-lr-markers parameter (on/off). This is intended to be used in SSA mode.
    default value: 'on'

  --wave-bandwidth (int32) = Wave correction bandwidth.
    default value: '101'

  --wave-bin-count (int32) = Wave correction bin count.
    default value: '25'

  --wave-count (int32) = Wave correction wave count.
    default value: '-1'

  --wave-smooth (bool) = Wave correction wave smooth.
    default value: 'True'

  --median-autosome-median-normalization (bool) = Log2Ratio Median Autosmome Median Normalization
    default value: 'True'

  --shrinkage-v2-node:enable (flag) = Enable shrinkage v2 node (default false)
    Aliases: --shrink-v2-enable

  --shrink-lprec (vec_float32) = Precision of the likelihood for wavelet shrinkage.
    default value: '0.250000,2.000000'

  --outlier-shrink-lprec (vec_float32) = Precision of the likelihood for wavelet shrinkage outlier handling.
    default value: '0.025000,2.000000,8.000000'

  --shrink-converge (float64) = Convergence criteria for shrinkage.
    default value: '0.000001'

  --outlier-shrink-converge (float64) = Convergence criteria for shrinkage when downweighting outliers.
    default value: '0.000001'

  --shrink-downweight-maxiter (int32) = Maximum iterations for downweighting outliers during shrinkage.
    default value: '25'

  --outlier-shrink-large-stdev (float64) = Large standard deviation for downweighting outliers in shrinkage.
    default value: '125.000000'

  --hmmCN_state (string8) = The state values for the CN HMM.
    default value: '0,1,2,3,4,5'

  --hmmCN_mu (string8) = The Mu values for the CN HMM.
    default value: '-1.63,-0.58,0,0.45,0.72,0.93'

  --hmmCN_sigma (string8) = The sigma values for the CN HMM.
    default value: '0.3,0.3,0.3,0.3,0.3,0.3'

  --diagonal-weight (float64) = The diagonal weight value for the CN HMM.
    default value: '0.995000'

  --mapd-weight (float64) = The mapd weight value for the CN HMM.
    default value: '0.220000'

  --min-segment-size (int32) = The minimum segment size value for the CN HMM.
    default value: '5'

  --hmm-confidence-weight (float64) = The confidence weight value for the CN HMM.
    default value: '0.600000'

  --hmmCN_state-X (string8) = The X chromosome state values for the CN HMM.
    default value: '0,1,2,3,4,5'

  --hmmCN_mu-X (string8) = The X chromosome Mu values for the CN HMM.
    default value: '-1.63,-0.58,0,0.45,0.72,0.93'

  --hmmCN_sigma-X (string8) = The X chromosome sigma values for the CN HMM.
    default value: '0.3,0.3,0.3,0.3,0.3,0.3'

  --diagonal-weight-X (float64) = The X chromosome diagonal weight value for the CN HMM.
    default value: '0.995000'

  --mapd-weight-X (float64) = The X chromosome mapd weight value for the CN HMM.
    default value: '0.220000'

  --min-segment-size-X (int32) = The X chromosome minimum segment size value for the CN HMM.
    default value: '5'

  --hmm-confidence-weight-X (float64) = The X chromosome HMM confidence weight value for the CN HMM.
    default value: '0.600000'

  --hmmCN_state-Y (string8) = The Y chromosome state values for the CN HMM.
    default value: '0,1,2,3,4,5'

  --hmmCN_mu-Y (string8) = The Y chromosome Mu values for the CN HMM.
    default value: '-1.63,-0.58,0,0.45,0.72,0.93'

  --hmmCN_sigma-Y (string8) = The Y chromosome sigma values for the CN HMM.
    default value: '0.3,0.3,0.3,0.3,0.3,0.3'

  --diagonal-weight-Y (float64) = The Y chromosome diagonal weight value for the CN HMM.
    default value: '0.995000'

  --mapd-weight-Y (float64) = The Y chromosome mapd weight value for the CN HMM.
    default value: '0.220000'

  --min-segment-size-Y (int32) = The Y chromosome minimum segment size value for the CN HMM.
    default value: '5'

  --hmm-confidence-weight-Y (float64) = The Y chromosome HMM confidence weight value for the CN HMM.
    default value: '0.600000'

  --hmm-shrink (bool) = Apply wavelet shrinkage to the log2 ratios.
    default value: 'True'

  --cn-gender-cutoff (float64) = CN Gender cutoff
    default value: '0.500000'

  --cn-state-threshold (float64) = CN Gender threshold for counting CN state as nonzero
    default value: '0.000000'

  --kernel-sigma-span (float64) = Probes spanned by one stdev.
    default value: '50.000000'

  --ap-ad-step (int32) = The step to use in calculating allele peaks.
    default value: '20'
    Aliases: --ap-step

  --ap-ad-window (int32) = The window to use in calculating allele peaks.
    default value: '100'
    Aliases: --ap-window

  --ap-ad-point-count (int32) = The point count to use in calculating allele peaks.
    default value: '128'
    Aliases: --ap-point-count

  --ap-ad-bandwidth (float64) = The bandwidth to use in calculating allele peaks.
    default value: '0.250000'
    Aliases: --ap-bandwidth

  --ap-ad-cutoff (float64) = The cutoff to use in calculating allele peaks.
    default value: '0.050000'
    Aliases: --ap-cutoff

  --ap-ad-threshold (float64) = The clean threshold to use in calculating allele peaks.
    default value: '0.350000'
    Aliases: --ap-threshold

  --ap-ad-height-threshold (float64) = The height threshold factor to use in calculating allele peaks.
    default value: '0.000000'
    Aliases: --ap-height-threshold

  --ap-ad-height-threshold-bound (float64) = The bounds between which the height threshold is applied.
    default value: '1024.000000'
    Aliases: --ap-height-threshold-bound

  --ap-ad-symmetry (bool) = The symmetry value to use in calculating allele peaks.
    default value: 'True'
    Aliases: --ap-symmetry

  --ap-ad-fld-weights (filename) = Name of file with FLD-derived weights.
    Aliases: --ap-fld-weights

  --loh-error-rate (float64) = The error rate to use in LOH processing.
    default value: '0.050000'

  --loh-beta (float64) = The beta to use in LOH processing.
    default value: '0.001000'

  --loh-alpha (float64) = The alpha to use in LOH processing.
    default value: '0.010000'

  --loh-separation (int32) = The separation to use in LOH processing.
    default value: '1000000'

  --loh-min-marker-count (int32) = The minimum marker count to use in LOH processing.
    default value: '10'

  --loh-no-call-threshold (float64) = The no call threshold to use in LOH processing.
    default value: '0.050000'

  --loh-min-genomic-span (int32) = The minimum genomic span to use in LOH processing.
    default value: '1000000'

  --gains-boundries (string8) = Mosaicism Gains Boundries
    default value: '0.08764945,0.15380349,0.21465931,0.27100300'

  --losses-boundries (string8) = Mosaicism Losses Boundries
    default value: '-0.08293345,-0.17551812,-0.28048196,-0.40165383'

  --marker-bandwidth (int32) = Mosaicism Marker Bandwitdth
    default value: '6000'

  --confidence-window (int32) = Mosaicism confidence running median window size
    default value: '251'

  --run-y-chromosome (bool) = Run Mosaicism analysis on Y Chromosome
    default value: 'True'

  --report-file (string8) = Report file name.

  --text-output (bool) = Output data in ASCII text format.
    default value: 'False'

  --oschp-output (bool) = Output data in HDF5 format.
    default value: 'False'
    Aliases: --hdf5-output

  --mangle-probeset-names (bool) = Prefix probe set names written to the CYCHP file with S- or C-
    default value: 'False'

  --meta-data-info (vec_string8) = Meta data in key=value pair format that will be written to the header.

  --waviness-block-size (int32) = marker count
    default value: '50'

  --waviness-genomic-span (int32) = genomic segment length
    default value: '0'

  --@help       = print developer help

</pre><p><a class="anchor" id="faq"></a> </p>
<h2>Frequently Asked Questions </h2>
<p><a class="anchor" id="references"></a> </p>
<h2>References </h2>
<ul>
<li>BRLMM-P - <a href="http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf">http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf</a></li>
<li>PLIER - <a href="http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf">http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf</a> </li>
</ul>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated for Affymetrix Power Tools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
