<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Affymetrix Power Tools: MANUAL: apt-copynumber-onco-ref (2.4.0)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="LICENSE.html"><span>License</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>MANUAL: apt-copynumber-onco-ref (2.4.0) </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction.</a></li>
<li><a href="#quickstart">Quick Start</a></li>
<li><a href="#options">Options</a></li>
<li><a href="#references">References</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p><b>apt-copynumber-onco-ref</b> will produce a <b>Reference Model File</b> which can then be used by apt-copynumber-onco-ssa for single-sample analysis of OncoScan experiments. Inputs consist of CEL files for a number of OncoScan experiments (each experiment consists of two CEL files, one for the AT and one for the GC channels), and analysis and annotation files.</p>
<p><code>apt-copynumber-onco-ref</code> requires a minimum of six experiments, including at least one male and one female sample, to run; however a mininum of 22 male and 22 female samples is recommended.</p>
<p><a class="anchor" id="quickstart"></a> </p>
<h2>Quick Start </h2>
<h3>Input/Output Files</h3>
<p>The following input files are required by <code>apt-copynumber-onco-ref</code>:</p>
<table class="doxtable">
<tr>
<th>Name </th><th>File extension/type </th><th>Description  </th></tr>
<tr>
<td>spf-file </td><td>spf5 (HDF5) </td><td>Describes probes and probesets on array. </td></tr>
<tr>
<td>annotation-file </td><td>annot.db (SQLite) </td><td>Provides annotation data for probesets. </td></tr>
<tr>
<td>x-probes-file </td><td>chrXprobes (text) </td><td>Probes for X chromosome (used in raw gender calling) </td></tr>
<tr>
<td>y-probes-file </td><td>chrYprobes (text) </td><td>Probes for Y chromosome (used in raw gender calling) </td></tr>
<tr>
<td>cel-pairs-file </td><td>text </td><td>Identifies the CEL file pairs (AT and GC channel per sample) for reference generation </td></tr>
<tr>
<td>covariates-file </td><td>refcovar (HDF5) </td><td>Probeset intensity covariates </td></tr>
<tr>
<td>probe-covariates-file </td><td>refcovar (HDF5) </td><td>Probe covariates file </td></tr>
<tr>
<td>snp-priors-input-file </td><td>text </td><td>SNP priors file </td></tr>
</table>
<p>The following output files are generated by apt-copynumber-ref:</p>
<table class="doxtable">
<tr>
<th>Name </th><th>File extension/type </th><th>Description  </th></tr>
<tr>
<td>reference-file </td><td>REF_MODEL (HDF5) </td><td>The reference model file for the copynumber probesets </td></tr>
<tr>
<td>snp-posteriors-output-file </td><td>txt </td><td>SNP posteriors file (optional) </td></tr>
<tr>
<td>snp-priors-output-file </td><td>txt </td><td>SNP priors file (optional) </td></tr>
<tr>
<td>sig-snp-report-file </td><td>txt </td><td>Results of signature SNP checks for the experiments. </td></tr>
</table>
<h3>CEL files lists:</h3>
<p>The CEL file pairs are specified in a tab delimited text file with two columns representing the AT and GC-channel CEL files respectively. The first row of the cel-pairs-file should contain the column names <code>atchannelcel</code> and <code>gcchannelcel</code>.</p>
<p>See <a href="VIGNETTE-OncoScan-ref.html">OncoScan reference model generation</a> for examples.</p>
<p><a class="anchor" id="options"></a> </p>
<h2>Options: </h2>
<pre class="fragment">
   'apt-copynumber-onco-ref'
   2.4.0 svn://svn.affymetrix.com/apt2-copynumber/branches/APT_CN_2_4_0_BRANCH@1018(1495) 2015-08-25-04:44

USER HELP ----------

  --version (flag) = Display the version info for this program.

  --user-help (flag) = Display help intended for end-users.
    Aliases: --help, -h

  --log-file (string8) = The name of the log file.  Defaults to 'PROGRAMNAME-NUMBER.log'.
    Aliases: --lf

  --console-add-select (string8) = Add selectors for console messages.
    e.g. to include all debug messages: --console-add-select debug
         to include all messages of level 3 or higher: --console-add-select '*:3'


  --console-add-neg-select (string8) = Add selectors to be excluded from console messages.
    e.g. to exclude all warning messages and errors summary: --console-add-neg-select WARNING,summary


  --arg-file (filename) = Read arguments from this file.
    File should be a valid XML file, and should contain elements of the form
    &lt;Parameter name="PARAMETER-NAME" analysis="NODE-NAME" currentValue="VALUE" /&gt;.

  --check-param (flag) = Stop the program after the command parameters have been checked.

  --out-dir (string8) = The output directory for result files.

  --temp-dir (string8) = The output directory for temporary files.

  --force (bool) = Force the files to process regardless of array type.
    default value: 'False'

  --keep-intermediate-data (bool) = Dump data on itermediate calculations for testing purposes.
    default value: 'False'

  --delete-temp-files (bool) = Delete the temporary files after the run completes.
    default value: 'True'

  --keep-combined-cels (bool) = Keep the combined cel file data.
    default value: 'False'

  --reference-file (string8) = The reference output file (binary format).

  --reference-file-text (string8) = The reference output file (text format).

  --brlmmp-MS (float64) = brlmmp MS. confidence threshold
    default value: '0.050000'

  --spf-file (string8) = The SPF input file.

  --annotation-file (string8,required) = The annotation input file.

  --x-probes-file (string8,required) = The file that identifies the X chromosome probe ids.

  --y-probes-file (string8,required) = The file that identifies the Y chromosome probe ids.

  --cel-pairs-file (string8) = The file that identifies the cel file pairs to process.

  --cel-files (string8) = The file that identifies the cel files to process.

  --covariates-file (string8) = The string that identifies the probe set covariates to load into the reference.

  --probe-covariates-file (string8) = The string that identifies the probe covariates to load into the reference.

  --fld-file (string8) = The string that identifies the pFLD values to load into the reference.

  --ap-baf-fld-weights (string8) = File of BAF FLD weights to be loaded into the reference alongside the pFLD values.
    Aliases: --baf-fld-weights

  --meta-data-info (vec_string8) = Meta data in key=value pair format that will be written to the CopyNumber/Parameters dataset.

  --paired-reference-file (filename) = The REF_MODEL file that is to be paired with the generated reference file.

  --icov-order (vec_string8) = Names of the intensity covariates in the order that they are to be applied.

  --icov-bin-type (vec_string8) = Bin type for each covariate. (equally-populated, equally-spaced, discrete)

  --icov-bin-count (vec_int32) = Bin count for each covariate.

  --icov-cn-percentile (vec_float32) = Percentile calculated per bin for each covariate taken on CN probe sets.

  --icov-snp-percentile (vec_float32) = Percentile calculated per bin for each covariate taken on SNP probe sets.

  --icov-channel-norm (vec_string8) = Indicates if channel normalization should be applied for each covariate (on, off)

  --icov-min-nd-probe-count-per-bin (int32) = Minimum number of normal diploid probes required for each bin.
    default value: '30'

  --icov-use-nd-signal-markers (string8) = Use normal diploid markers (on, off).
    default value: 'off'

  --icov-disable (bool) = Use this to disable intensity covariate adjustment. Removing all icov-* parameters will have the same effect
    default value: 'False'

  --lrcov-order (vec_string8) = Names of the log2 ratio covariates in the order that they are to be applied.

  --lrcov-bin-type (vec_string8) = Bin type for each covariate. (equally-populated, equally-spaced, discrete)

  --lrcov-bin-count (vec_int32) = Bin count for each covariate.

  --lrcov-iqr-scaling (vec_string8) = Inidcates if IQR scaling is on or off for each covariate (on, off).

  --lrcov-subtract-from-XY (vec_string8) = Indicates if bin median is to be subracted from X-Y probe sets (on, off).

  --lrcov-min-nd-marker-count-per-bin (int32) = Minimum number of normal diploid probe sets required for each bin.
    default value: '30'

  --lrcov-use-nd-lr-markers (string8) = Use normal diploid markers (on, off).
    default value: 'on'

  --lrcov-disable (bool) = Use this to disable log2 ratio covariate adjustment. Removing all lrcov-* parameters will have the same effect
    default value: 'False'

  --apcov-order (vec_string8) = Names of the allele peaks covariates in the order that they are to be applied.

  --apcov-bin-type (vec_string8) = Bin type for each covariate. (equally-populated, equally-spaced, discrete)

  --apcov-bin-count (vec_int32) = Bin count for each covariate.

  --apcov-coarse-allele-peak-adjustment (vec_string8) = Apply coarse allele peak adjustment for each covariate (on, off).

  --apcov-coarse-allele-peak-adjustment-step (vec_int32) = 

  --apcov-coarse-allele-peak-adjustment-window (vec_int32) = 

  --apcov-coarse-allele-peak-adjustment-point-count (vec_int32) = 

  --apcov-coarse-allele-peak-adjustment-bandwidth (vec_float32) = 

  --apcov-coarse-allele-peak-adjustment-cutoff (vec_float32) = 

  --apcov-coarse-allele-peak-adjustment-clean-threshold (vec_float32) = 

  --apcov-coarse-allele-peak-adjustment-outlier-trim (vec_float32) = 

  --apcov-covariate-bin-peaks-point-count (int32) = 
    default value: '0'

  --apcov-covariate-bin-peaks-bandwidth (float64) = 
    default value: '0.000000'

  --apcov-master-peaks-point-count (int32) = 
    default value: '0'

  --apcov-master-peaks-bandwidth (float64) = 
    default value: '0.000000'

  --apcov-kernel-function-selection (string8) = 

  --apcov-disable (bool) = Use this to disable allele-peak covariate adjustment. Removing all apcov-* parameters will have the same effect
    default value: 'False'

  --agr-denominator-source (string8) = The denominator to use in calculating the antigenomic ratio.
    default value: 'SNP'

  --agr-denominator-percentile (float64) = The percentile denominator to use in calculating the antigenomic ratio denominator.
    default value: '0.750000'

  --igender-female-threshold (float64) = The female threshold to use in the raw intensity gender call.
    default value: '0.440000'

  --igender-male-threshold (float64) = The male threshold to use in the raw intensity gender call.
    default value: '0.630000'

  --nddicov-min-nd-probe-count-per-bin (int32) = Override the icov-min-nd-probe-count-per-bin parameter. This is intended to be used in SSA mode.
    default value: '30'

  --nddicov-use-nd-signal-markers (string8) = Override the icov-use-nd-signal-markers parameter (on/off). This is intended to be used in SSA mode.
    default value: 'off'

  --sketch-size (int32) = The sketch size to use.
    default value: '50000'

  --dual-channel-normalization (bool) = Use dual channel normalization.
    default value: 'False'

  --snp-priors-input-file (string8) = The string that identifies the snp priors to use when genotyping.

  --snp-posteriors-output-file (string8) = The string that identifies the snp posteriors to output when genotyping.

  --snp-priors-output-file (string8) = The string that identifies the snp priors to output when genotyping.

  --brlmmp-HARD (int32) = brlmmp hardshell
    default value: '3'

  --brlmmp-SB (float64) = brlmmp shellbarrier
    default value: '0.450000'

  --brlmmp-CM (int32) = brlmmp callmethod (1=batch mode, 2=single sample mode)
    default value: '1'

  --brlmmp-bins (int32) = brlmmp bins
    default value: '100'

  --brlmmp-mix (int32) = brlmmp mix
    default value: '1'

  --brlmmp-bic (float64) = brlmmp bic
    default value: '2.000000'

  --brlmmp-CSepPen (float64) = brlmmp CSepPen
    default value: '0.000000'

  --brlmmp-CSepThr (float64) = brlmmp CSepThr
    default value: '16.000000'

  --brlmmp-lambda (float64) = brlmmp lambda
    default value: '1.000000'

  --brlmmp-wobble (float64) = brlmmp wobble
    default value: '0.050000'

  --brlmmp-copyqc (float64) = brlmmp copyqc. 0 = no test for copy qc
    default value: '0.000000'

  --brlmmp-copytype (int32) = brlmmp copytype. 0 = standard copy qc method
    default value: '0'

  --brlmmp-ocean (float64) = brlmmp ocean
    default value: '0.000000'

  --brlmmp-clustertype (int32) = brlmmp clustertype. 1 = standard 1-d clustering only
    default value: '1'

  --brlmmp-transform (string8) = brlmmp transform. (mva, rvt, ssf, ces, assf, ccs)
    default value: 'mva'

  --brlmmp-hints (int32) = brlmmp hints
    default value: '0'

  --brlmmp-IsoHetY (float64) = brlmmp IsoHetY
    default value: '0.000000'

  --brlmmp-inflatePRA (float64) = brlmmp inflatePRA. 0 = no increase in uncertainty
    default value: '0.000000'

  --brlmmp-CP (float64) = brlmmp contradiction penalty. 0 = no penalty
    default value: '0.000000'

  --sig-snp-min-call-rate (float64) = Threshhold for minimum allowable call rate per channel for signature snp probesets
    default value: '0.600000'

  --sig-snp-min-compare-rate (float64) = Threshhold for minimum percentage of paired probesets with valid AT and GC calls
    default value: '0.650000'

  --sig-snp-min-concordance (float64) = Threshhold for minimum percentage of matching valid genotype calls
    default value: '0.600000'

  --sig-snp-min-strength-sketch-rank (float64) = Signal strength threshold under which signature snps will be considered a NoCall when calculating concordance
    default value: '0.305000'

  --sig-snp-report-filename (string8) = Name of optional output text file for signature snp report.

  --sig-gender-reference-chromosome (int32) = XY signal gender reference chromosome
    default value: '2'

  --sig-gender-xx-cutoff (float64) = XY signal gender XX cutoff
    default value: '0.850000'

  --sig-gender-xx-cutoff-high (float64) = XY signal gender XX cutoff high
    default value: '1.220000'

  --sig-gender-y-cutoff (float64) = XY signal gender Y cutoff
    default value: '0.310000'

  --nddlrcov-min-nd-marker-count-per-bin (int32) = Override the lrcov-min-nd-marker-count-per-bin parameter. This is intended to be used in SSA mode.
    default value: '30'

  --nddlrcov-use-nd-lr-markers (string8) = Override the lrcov-use-nd-lr-markers parameter (on/off). This is intended to be used in SSA mode.
    default value: 'on'

  --wave-trim (float64) = The trim to apply to the log2 ratios.
    default value: '2.000000'

  --wave-percentile (float64) = The high percentile value to apply.
    default value: '0.750000'

  --wave-demean (bool) = Demean the input to the SVD.
    default value: 'False'

  --wave-count (int32) = The number of waves to calculate.
    default value: '6'

  --@help       = print developer help

</pre><p><a class="anchor" id="faq"></a> </p>
<h2>Frequently Asked Questions </h2>
<p><a class="anchor" id="references"></a> </p>
<h2>References </h2>
<ul>
<li>BRLMM-P - <a href="http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf">http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf</a></li>
<li>PLIER - <a href="http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf">http://media.affymetrix.com/support/technical/technotes/plier_technote.pdf</a> </li>
</ul>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated for Affymetrix Power Tools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
