<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Affymetrix Power Tools: Affymetrix OncoScan OSCHP Data Format (2.4.0)</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="LICENSE.html"><span>License</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>Affymetrix OncoScan OSCHP Data Format (2.4.0) </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><h2>Contents </h2>
<ul>
<li><a href="#intro">Introduction.</a></li>
<li><a href="#format">Format</a><ul>
<li><a href="#header">Header Data</a></li>
<li><a href="#chromosome">Chromosome Data Group</a></li>
<li><a href="#probesets">ProbeSets Data Group</a></li>
<li><a href="#segments">Segments Data Group</a></li>
<li><a href="#genotyping">Genotyping Data Group</a></li>
<li><a href="#som">Somatic Mutation Data Group</a></li>
</ul>
</li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h2>Introduction </h2>
<p>OSCHP files contain analysis results generated by the APT OncoScan analysis pipeline and tools or by the OncoScan Console software. This document is intended to accompany one or more sample OSCHP files and to provide guidance for interpreting the data in the files. It is recommend that an HDF5 viewing tool, such as HDFView (<a href="http://www.hdfgroup.org/hdf-java-html/hdfview/index.html">http://www.hdfgroup.org/hdf-java-html/hdfview/index.html</a>) be used to inspect the contents of the files.</p>
<p><a class="anchor" id="format"></a> </p>
<h2>Format </h2>
<p>The format of the OSCHP files is the HDF5 binary format. Data is organized into a number of groups with each group containing zero or more sub-groups and zero or more datasets. Each sub-group may, in turn, contain zero or more sub-sub-groups and/or datasets, yielding a nested tree structure. Each dataset may have one or more columns or fields, and multiple rows. Consequently every dataset group or dataset can be described by a unique path in the file, such as /Group1/Group2/Dataset.</p>
<p>Full details of all the features of the HDF5 format are outside the scope of this document (see <a href="http://www.hdfgroup.org/HDF5/doc/index.html">http://www.hdfgroup.org/HDF5/doc/index.html</a> for more details).</p>
<p>The tree structure of the HDF5 files is as follows (datasets are in <b>bold</b>):</p>
<ul>
<li>Dset_IO_HDF5_Gdh (General Header Data)<ul>
<li><b>_&amp;keyvals</b> (data source: affymetrix-multi-data-type-analysis)</li>
<li>Dset_IO_HDF5_Gdh:0 (AT channel array)<ul>
<li><b>_&amp;keyvals</b> (data source: affymetrix-calvin-intensity)</li>
<li>Dset_IO_HDF5_Gdh:0:0<ul>
<li><b>_&amp;keyvals</b> (datasource: affymetrix-calvin-scan-acquisition)</li>
<li>Dset_IO_HDF5_Gdh:0:0:0<ul>
<li><b>_&amp;keyvals</b> (datasource: affymetrix-calvin-array)</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li>Dset_IO_HDF5_Gdh:1 (GC channel array)<ul>
<li><b>_&amp;keyvals</b> (data source: affymetrix-calvin-intensity)</li>
<li>Dset_IO_HDF5_Gdh:1:0<ul>
<li><b>_&amp;keyvals</b> (datasource: affymetrix-calvin-scan-acquisition)</li>
<li>Dset_IO_HDF5_Gdh:1:0:0<ul>
<li><b>_&amp;keyvals</b> (datasource: affymetrix-calvin-array)</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li>AlgorithmData (Data produced during algorithm development)<ul>
<li><b>MarkerABSignal</b> (internal use only)</li>
</ul>
</li>
<li>Chromosomes (Summary analysis results for each chromosome)<ul>
<li><b>Summary</b></li>
</ul>
</li>
<li>Genotyping (Genotyping analysis results from BRLMM-P)<ul>
<li><b>Calls</b></li>
<li><b>SigSnpReport</b></li>
</ul>
</li>
<li>ProbeSets (Analysis result for each probeset)<ul>
<li><b>AllelicData</b></li>
<li><b>CopyNumber</b></li>
</ul>
</li>
<li>Segments (Analysis results for segments)<ul>
<li><b>TuscanCN</b></li>
<li><b>TuscanUnmergedCN</b> (internal use only)</li>
<li><b>LOH</b></li>
</ul>
</li>
<li>SomaticMutation<ul>
<li><b>SomCall</b></li>
</ul>
</li>
</ul>
<p>Note that the structure is an extension of the existing structure used in Affymetrix CYCHP and CNCHP file formats.</p>
<p><a class="anchor" id="header"></a> </p>
<h2>Header Data </h2>
<p>The <code>Dset_IO_HDF5_Gdh</code> group contains header data in the form of <em>key/value</em> pairs. The groups structure is based on the General Data Header, or GDH, section of the Affymetrix Command Console binary format. The group contains a series of nested subgroups named <code>Dset_IO_HDF5_Gdh:&lt;level1&gt;:…:&lt;levelN&gt;</code> each of which may contain a dataset named <code>_&amp;keyvals</code>.</p>
<p>Specifically, for a pair of OncoScan arrays, the group <code>Dset_IO_HDF5_Gdh</code> will contain two sub-groups <code>Dset_IO_HDF5_Gdh:0</code> and <code>Dset_IO_HDF5_Gdh:1</code>, and a dataset _&amp;keyvals. The sub-groups <code>/Dset_IO_HDF5_Gdh/Dset_IO_HDF5_Gdh:0</code> and <code>/Dset_IO_HDF5_Gdh/Dset_IO_HDF5_Gdh:1</code> will contain GDH data copied from the CEL files for the AT and the GC arrays respectively, including data generated by Affymetrix Command Console and upstream analysis steps (see .CEL file documentation for details), while the dataset <code>/Dset_IO_HDF5_Gdh/_&amp;keyvals</code> will contain summary and metadata generated by the APT OncoScan pipeline. For a tumor experiment as part of a matched normal/tumor pair (i.e. set of four CEL files), the group <code>Dset_IO_HDF5_Gdh</code> will contain <em>four</em> subgroups, corresponding to the GDH data for all four CEL files.</p>
<p>Each row of a <code>_&amp;keyvals</code> dataset contains data about one key/value pair. The datasets contain three fields: key (the key of the pair), val (a string representing the value of the pair), and type (a string representing the type of the value, e.g. “string8” for a string of 8-bit characters, “int32” for a 32-bit integer, or “float32” for a 32-bit float).</p>
<p>Each key/value pair contains summary information or details of the program version and analysis parameters. The first key/value pair of each <code>_&amp;keyvals</code> dataset will have the key “data-source” and indicate the origin of the GDH group (this is used in mapping back to the Affymetrix Command Console binary format). The row with key “data-source” in the <code>/Dset_IO_HDF5_Gdh/_&amp;keyvals</code> will have the value “affymetrix-multi-data-type-analysis”.</p>
<p>Summary statistics begin with the prefix <code>affymetrix-chipsummary-</code> while analysis parameters begin with the prefix <code>affymetrix-algorithm-param-</code>.</p>
<p>Summary statistics are all in the key/values dataset with path <code>/Dset_IO_HDF5_Gdh/_&amp;keyvals</code> and included the following keys (<code>affymetrix-chipsummary-</code> has been omitted in the table):</p>
<table class="doxtable">
<tr>
<th>Key</th><th>Description  </th></tr>
<tr>
<td>raw-intensity-ratio-gender</td><td>Result of raw-intensity gender calling. Either “male”, “female” or “unknown” </td></tr>
<tr>
<td>raw-intensity-ratio</td><td>Raw intensity gender ratio: mean(raw intensity for Y probes) / mean(raw intensity for X probes) (Internal use only) </td></tr>
<tr>
<td>XX-ratio</td><td>median(signal for X probes) / median(median ref signal from ref chromosome). (Internal use only) </td></tr>
<tr>
<td>Y-ratio</td><td>median(signal for Y probes) / median(median ref signal from ref chromosome) (Internal use only) </td></tr>
<tr>
<td>XX</td><td>1 if XX-Ratio lies between xx-cutoff and xx-cutoff-high; 0 otherwise (where xx-cutoff and xx-cutoff-high are analysis parameters) </td></tr>
<tr>
<td>Y</td><td>1 if Y-Ratio is greater than y-cutoff; 0 otherwise (where y-cutoff is an analysis parameter) </td></tr>
<tr>
<td>Y-gender-call-computed</td><td>1 if Y-gender call was computed. 0 otherwise. </td></tr>
<tr>
<td>Y-gender-call</td><td>Y-chromosome CN gender call. Either “male” or “female”. Male if Y-gender-ratio is above cutoff. </td></tr>
<tr>
<td>Y-gender-not-zero</td><td>Number of probes from Y-chromosome probes list with non-zero CN (Internal use only) </td></tr>
<tr>
<td>Y-gender-Y-count</td><td>Number of Y-chromosome probes for calculating Y-gender call. (A subset of all probes on Y-chromosome can be specified for gender calling.) (Internal use only) </td></tr>
<tr>
<td>Y-gender-confidence</td><td>Confidence scored for Y-gender call </td></tr>
<tr>
<td>Y-gender-ratio</td><td>(Y-gender-not-zero / Y-gender-Y-count) (Internal use only) </td></tr>
<tr>
<td>MedianAutosomeMedian</td><td>Median of median log2 ratios for all autosomes </td></tr>
<tr>
<td>MAPD</td><td>Median Absolute Pairwise Difference of log2 ratios </td></tr>
<tr>
<td>iqr</td><td>Interquartile range of log2 ratios for all chromosomes </td></tr>
<tr>
<td>call-rate</td><td>Proportion of genotypable SNPs that have a genotype call </td></tr>
<tr>
<td>wave-count</td><td>Number of waves used in Log2 ratio wave correction </td></tr>
<tr>
<td>sample-hom-frequency</td><td>The frequency (homozygous calls / all SNP calls) of SNP homozygous calls for the sample </td></tr>
<tr>
<td>sample-het-frequency</td><td>The frequency (heterozygous calls / all SNP calls) of SNP heterozygous calls for the sample </td></tr>
<tr>
<td>median-raw-intensity</td><td>Median raw intensity of all probes from both of the AT chip and GC chip </td></tr>
<tr>
<td>Q3-raw-intensity</td><td>The 75th percentile of raw intensities among probes used for the analysis from both of the AT chip and GC chip </td></tr>
<tr>
<td>Q3-raw-intensity-AT</td><td>The 75th percentile of raw intensities among probes used for the analysis from AT chip </td></tr>
<tr>
<td>Q3-raw-intensity-GC</td><td>The 75th percentile of raw intensities among probes used for the analysis from GC chip </td></tr>
<tr>
<td>AGR</td><td>The ratio of median raw intensities of antigenomic probes on both arrays to Q3-raw-intensity </td></tr>
<tr>
<td>AGR-AT</td><td>The ratio of median raw intensities of antigenomic probes on AT array to Q3-raw-intensity-AT </td></tr>
<tr>
<td>AGR-GC</td><td>The ratio of median raw intensities of antigenomic probes on GC array to Q3-raw-intensity-GC </td></tr>
<tr>
<td>snp-qc-algorithm</td><td>Indicates the SNPQC algorithm used. The value will be “EMQC” for OncoScan </td></tr>
<tr>
<td>snp-qc</td><td>SCAR-based QC metric for SNP performance computed based on the SNPs defined in the snp list </td></tr>
<tr>
<td>snp-qc-type</td><td>Either “ND” or “non-ND”. Indicates if nd-snp-qc was calculated using Normal Diploid markers or not. If “non-ND” then nd-snp-qc is set to the same value as snp-qc. </td></tr>
<tr>
<td>nd-snp-qc</td><td>Similar to snp-qc but based on Normal Diploid SNPs if sufficient number are detected. </td></tr>
<tr>
<td>raw-snp-qc</td><td>Similar to snp-qc except for the input is log2(median raw intensity of A allele/ median raw intensity of B allele) computed based on SNP probes defined in the snp list </td></tr>
<tr>
<td>nd-raw-snp-qc</td><td>Similar to raw-snp-qc except for the markers are from normal diploid SNPs identified by the normal diploid algorithm </td></tr>
<tr>
<td>cel-pair-check-status</td><td>The result of Signature SNP analysis for the two CEL files (AT and GC channels) used for experiments (<code>Pass</code>, <code>LikelyCELmispair</code>, <code>Warn|LikelyGCinATchannel</code>, etc.) </td></tr>
<tr>
<td>cel-pair-check-at-call-rate</td><td>Percentage of signature SNPs for which calls were generated on AT channel </td></tr>
<tr>
<td>cel-pair-check-gc-call-rate</td><td>Percentage of signature SNPs for which calls were generated on GC channel </td></tr>
<tr>
<td>cel-pair-check-compare-rate</td><td>Percentage of signature SNPs used in comparison </td></tr>
<tr>
<td>cel-pair-check-concordance</td><td>Concordance value for signature SNPs across AT/GC channels </td></tr>
<tr>
<td>genome-LOH</td><td>Fraction of genome reporting LOH </td></tr>
<tr>
<td>autosome-genome-LOH</td><td>Fraction of autosomal genome reporting LOH.  </td></tr>
<tr>
<td>waviness-seg-count-loss</td><td>Number of autosomal segments containing more than 50 markers that have copy number calls are less than 2 </td></tr>
<tr>
<td>waviness-seg-count-gain</td><td>Number of autosomal segments containing more than 50 markers that have copy number calls are greater than 2 </td></tr>
<tr>
<td>waviness-seg-count</td><td>Number of autosomal segments containing more than 50 markers that have copy number calls are not 2 </td></tr>
<tr>
<td>waviness-sd</td><td>The residual standard deviation (SD) after correcting for adjacent probe set to probe set SD based on autosomal log2 ratios </td></tr>
<tr>
<td>nd-waviness-sd</td><td>The residual standard deviation (SD) after correcting for adjacent probe set to probe set SD based on normal-diploid markers only </td></tr>
<tr>
<td>waviness-amplitudes</td><td>Coefficients of the estimated waves that are applied to the log2 ratios </td></tr>
<tr>
<td>log2-ratio-gradient</td><td>The standard deviation of low pass filter values used in image gradient correction, multiplied by 128 in order to bring the value found on most chips into the range of 1 to 10 </td></tr>
<tr>
<td>percent-nd</td><td>Percent of normal diploid markers for autosomes </td></tr>
<tr>
<td>number-nd</td><td>Total number of normal diploid markers for autosomes </td></tr>
<tr>
<td>tuscan-ploidy</td><td>Ploidy or dominating copynumber state (calculated as weighted CN median across all autosomes) reported by TuScan </td></tr>
<tr>
<td>tuscan-percent-tumor</td><td>Percent tumor reported by TuScan </td></tr>
<tr>
<td>tuscan-percent-AC</td><td>Aberrant Cell Percent – values are either “NA” if tumor burden cannot be calculated, “homogeneous” if percent tumor burden is greater than 95%, or percent tumor burden rounded to the nearest 5. </td></tr>
<tr>
<td>tuscan-percent-ACDC</td><td>Aberrant Cell Derived Copynumber – either “Yes” or “No” – indicates if TuScan algorithm was able to calculate a copy number call from percent tumor burden </td></tr>
<tr>
<td>tuscan-nGoF4</td><td>Goodness-of-fit reported by TuScan </td></tr>
<tr>
<td>tuscan-normalized-nGoF</td><td>Goodness-of-fit reported by TuScan </td></tr>
<tr>
<td>tuscan-recentering-occured</td><td>Either "Yes" or "No" - indicates whether TuSCan detected that log2ratio of 0 did not correspond to 2 copies and adjusted accordingly </td></tr>
<tr>
<td>tuscan-l2r-adjustment</td><td>The offset applied to log2 rations if recentering was applied </td></tr>
<tr>
<td>tuscan-reliability-score</td><td>Measure of goodness of fit between actual and predicted log2 ratio and BAF. Values generally in the range 0 to 1.1. Value may be "NA" if TuScan ploidy reports "NA" and in some cases where Aberrant Cells Percent reports "homogeneous". </td></tr>
<tr>
<td>Exception-LowDiploid</td><td>An exception reported if there are insufficient Normal Diploid probesets identified. Values 1/0 </td></tr>
<tr>
<td>low-diploid-flag</td><td>An exception reported if there are insufficient Normal Diploid probesets identified. Values: “Yes”/“No”.  </td></tr>
<tr>
<td>SNR</td><td>Signal/noise ratio (Internal use only) </td></tr>
<tr>
<td>SNR-AT</td><td>Signal/noise ratio of AT channel (Internal use only) </td></tr>
<tr>
<td>SNR-GC</td><td>Signal/noise ratio of GC channel (Internal use only) </td></tr>
<tr>
<td>ndSNR</td><td>Signal/noise ratio for ND probesets (Internal use only) </td></tr>
<tr>
<td>ndSNR-AT</td><td>Signal/noise ratio for ND probesets for AT channel (Internal use only) </td></tr>
<tr>
<td>ndSNR-GC</td><td>Signal/noise ratio for ND probesets for GC channel (Internal use only) </td></tr>
<tr>
<td>hyb-control-raw-signal-AT</td><td>Hybridization control raw signal metric for AT channel </td></tr>
<tr>
<td>hyb-control-raw-signal-GC</td><td>Hybridization control raw signal metric for GC channel </td></tr>
<tr>
<td>somHighCount</td><td>Total number of High Confidence Somatic Mutation probesets </td></tr>
<tr>
<td>somLowCount</td><td>Total number of Low Confidence SomaticMutation probesets </td></tr>
<tr>
<td>somMedianMutationScore</td><td>Median Somatic Mutation score value </td></tr>
</table>
<p><a class="anchor" id="chromosome"></a> </p>
<h2>Chromosome Data Group </h2>
<p>The <b>Chromosomes</b> data group consists of a single data set called <b>Summary</b>. The Summary data set contains summary information for each chromosome. This data set will contain the following columns:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Column Type </th><th>Description  </th></tr>
<tr>
<td>Chromosome </td><td>UByte </td><td>The values in this column are defined as 1-22, 24 (for X), 25 (for Y) and 26 (for MT) and 255 (for no value). The chromosome display values (1-22, X, Y, MT) are stored in the Display field. </td></tr>
<tr>
<td>Display </td><td>Text </td><td>label for the chromosome. </td></tr>
<tr>
<td>StartIndex </td><td>UInt </td><td>Zero-based reference to the row in the CopyNumber data set of the ProbeSets data group containing results for this chromosome </td></tr>
<tr>
<td>MarkerCount </td><td>UInt </td><td>The number of markers in the chromosome </td></tr>
<tr>
<td>MinSignal </td><td>Float </td><td>Minimum log2 ratio value found in the chromosome* </td></tr>
<tr>
<td>MaxSignal </td><td>Float </td><td>Maximum log2 ratio value found in the chromosome* </td></tr>
<tr>
<td>MedianCnState </td><td>Float </td><td>Median copynumber state reported by TuScan </td></tr>
<tr>
<td>HomFrequency </td><td>Float </td><td>Frequency of Homozygosity </td></tr>
<tr>
<td>HetFrequency </td><td>Float </td><td>Frequency of Heterozygosity </td></tr>
<tr>
<td>LOH </td><td>Float </td><td>Proportion of markers reporting LOH calls per chromosome </td></tr>
<tr>
<td>MedianSignal </td><td>Float </td><td>Median log2 ratio value found in the chromosome* </td></tr>
</table>
<p>The log2 ratios are trimmed log2 ratios by default, but the values might go beyond the trimmed values due to the GC correction and/or wave correction steps.</p>
<p><a class="anchor" id="probesets"></a> </p>
<h2>ProbeSets Data Group </h2>
<p>The <b>Probesets</b> data group consists of two data sets, called <b>AllelicData</b> and <b>CopyNumber</b> containing analysis results specific to the probesets.</p>
<p>The data rows for each chromosome are consecutive, in a single block, starting at the index given in the "Chromosomes/Summary" data set in the StartIndex column. The number of data rows for each chromosome is exactly equal to the value given in the "Chromosomes/Summary" data set in the MarkerCount column.</p>
<p>The <b>CopyNumber</b> data set will contain the following columns:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Column Type </th><th>Description  </th></tr>
<tr>
<td>ProbeSetName </td><td>Ascii </td><td>The name of the probeset. The probeset names begin with "C" (copy number probes) or "S" (SNP probes). </td></tr>
<tr>
<td>Chromosome </td><td>UByte </td><td>The values in this column are defined as 1-22, 24 (for X), 25 (for Y) and 26 (for MT) and 255 (for no value). (Corresponding chromosome display values (1-22, X, Y, MT) are stored in /Chromosomes/Summary dataset). </td></tr>
<tr>
<td>Position </td><td>UInt </td><td>Genomic position in the chromosome. The Position values for each chromosome are strictly increasing. Each position can be assosciated with one probe only. </td></tr>
<tr>
<td>Log2Ratio </td><td>Float </td><td>Log2 ratio (signal / median of Reference signal) </td></tr>
<tr>
<td>WeightedLog2Ratio </td><td>Float </td><td>Signal restored log2 ratio value </td></tr>
<tr>
<td>SmoothSignal </td><td>Float </td><td>Each value is the result of a kernel smooth over a window of genomically contiguous log2 ratios. By default this is converted to a copy number value using a log-linear model that fits each CN state to its expected log2 ratio. This is useful for seeing mosaics. (Internal use only) </td></tr>
<tr>
<td>NormalDiploid </td><td>UInt </td><td>1 = diploid, 0 = non-diploid (Internal use only) </td></tr>
</table>
<p>The <b>AllelicData</b> data set houses data summarizing allele patterns and contains the following columns:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Column Type </th><th>Description  </th></tr>
<tr>
<td>ProbesetName </td><td>Ascii </td><td>The name of the probeset </td></tr>
<tr>
<td>Chromosome </td><td>UByte </td><td>Chromosome number, as for CopyNumber data set </td></tr>
<tr>
<td>Position </td><td>UInt </td><td>Genomic position in the chromosome </td></tr>
<tr>
<td>AllelicDifference </td><td>Float </td><td>Allelic difference value </td></tr>
<tr>
<td>BAF </td><td>Float </td><td>B-allele frequency (range 0.0-1.0) </td></tr>
</table>
<p><a class="anchor" id="segments"></a> </p>
<h2>Segments Data Group </h2>
<p>The Segments data group contains data sets housing segments data. The Segments group currently contains the following datasets:</p>
<ul>
<li><b>TuscanCN</b> – The merged segments reported by the TuScan algorithm.</li>
<li><b>LOH</b> - Segments reported by the Loss of Heterozygosity algorithm.</li>
<li><b>TuScanUnmergedCN</b> – <em>This table is for internal use only</em>.</li>
</ul>
<p>Both the <b>TuscanCN</b> and the <b>LOH-Segments</b> data sets in this data group will contain the following columns:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Column Type </th><th>Description  </th></tr>
<tr>
<td>SegmentID </td><td>UInt </td><td>Segment identifier unique to the OSCHP file </td></tr>
<tr>
<td>Chromosome </td><td>UByte </td><td>The chromosome display values (1-22, X, Y, MT) are stored in the data set header. The values in this column are defined as 1-22, 24 (for X), 25 (for Y) and 26 (for MT) and 255 (for no value) </td></tr>
<tr>
<td>StartPosition </td><td>UInt </td><td>Start genomic position of the segment. The StartPosition column is strictly increasing within each dataset. </td></tr>
<tr>
<td>StopPosition </td><td>UInt </td><td>Ending genomic position of the segment. The StopPosition is always greater than the StartPosition in the same row and less than the StartPosition in the next row. </td></tr>
<tr>
<td>MarkerCount </td><td>Int </td><td>Number of markers in the segment. </td></tr>
</table>
<p>In addition to the columns above, <b>TuscanCN</b> data set contains the following columns:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Column Type </th><th>Description  </th></tr>
<tr>
<td>TotalCN </td><td>Float </td><td>nA + int nB or fraction in AlternateMode </td></tr>
<tr>
<td>BinnedCN </td><td>String </td><td>The representative value for the bin into which the CN value falls. (If the representative value is “as-is” then the actual value, rounded to a given number of decimal places, is shown). </td></tr>
<tr>
<td>MedianLog2ratio </td><td>Float </td><td>Median log 2 ratio </td></tr>
<tr>
<td>MedianBAF </td><td>Float </td><td>Median of markers identified as heterozygous, after mirroring any marker BAFs above 0.5 to the equivalent value below 0.5. It is NaN if the number of heterozygous markers in the segment is below 10 or the percent of homozygous markers is above 85% </td></tr>
</table>
<p>The <b>LOH</b> data set contains the following additional columns:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Column Type </th><th>Description  </th></tr>
<tr>
<td>MeanMarkerDistance </td><td>UInt </td><td>Mean distance between markers in the segment </td></tr>
<tr>
<td>LOH </td><td>UInt </td><td>The value of this column is 1 when a Loss of Heterozygosity is found, 0 when not found. </td></tr>
<tr>
<td>Confidence </td><td>Float </td><td>The ratio of the probability of the SCAR measurements under the LOH model to the sum of the probability under each of the LOH and non-LOH models. </td></tr>
</table>
<p><a class="anchor" id="genotyping"></a> </p>
<h2>Genotyping Data Group </h2>
<p>The <b>Genotyping</b> data group contains two data set, <b>Calls</b>, capturing data from the BRLMM-P genotyping algorithm, and <b>SigSnpReport</b>, capturing the results of the signature SNP analysis of the AT and CG channels.</p>
<p>The <b>Calls</b> data set will contain the following columns:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Column Type </th><th>Description  </th></tr>
<tr>
<td>Index </td><td>UInt </td><td>This column contains the index relative to the probeset in the CopyNumber data set of the ProbeSet data group </td></tr>
<tr>
<td>ProbeSetName </td><td>Ascii </td><td>The name of the probeset. </td></tr>
<tr>
<td>Call </td><td>UInt </td><td>Genotype calls (any ForcedCall that does not meet a given genotype confidence threshold will result in no call) </td></tr>
<tr>
<td>Confidence </td><td>Float </td><td>Genotype confidence </td></tr>
<tr>
<td>ForcedCall </td><td>UInt </td><td>Genotype calls without restriction to any genotype confidence </td></tr>
<tr>
<td>ASignal </td><td>Float </td><td>Summarized signal for A allele </td></tr>
<tr>
<td>BSignal </td><td>Float </td><td>Summarized signal for B allele </td></tr>
<tr>
<td>SignalStrength </td><td>Float </td><td>0.5* (log2 A signal + log2 B signal) </td></tr>
<tr>
<td>Contrast </td><td>Float </td><td>log2 B signal – log2 A signal </td></tr>
</table>
<p>The <b>SigSnpReport</b> data set contains data from the genotyping results of a set of signature SNPs which are used in checking that the AT and GC channel CEL files are correctly matched and classified. (Note the fields of this dataset follow the format of the Signature SNP report file.) The dataset will contain the following columns:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Column Type </th><th>Description  </th></tr>
<tr>
<td>Filename </td><td>String </td><td>The OSCHP file for the experiment </td></tr>
<tr>
<td>CEL filename </td><td>String </td><td>The name of one of the CEL files in the CEL file pair for the experiment </td></tr>
<tr>
<td>Channel </td><td>String </td><td>The channel for the CEL file: either “AT” or “GC” </td></tr>
<tr>
<td>CelPairCheckStatus </td><td>String </td><td>The result of the checking for the signature snps: either “Pass” or a string indicating the likely source of problems. </td></tr>
<tr>
<td>CelPairCheckCallRate </td><td>Float </td><td>The percent of signature SNPs for which calls were generated </td></tr>
<tr>
<td>CelPairCheckCompareRate </td><td>Float </td><td>The percent of signature SNPs used in comparing the two CEL files </td></tr>
<tr>
<td>CelPairConcordance </td><td>Float </td><td>The concordance for the signature SNPs across the two CEL files. </td></tr>
<tr>
<td>SIG_nnn (where nnn is an integer) </td><td>String </td><td>For signature SNP nnn, the genotyping call. Either “AA”, “AB”, “BB” or “NN”. </td></tr>
</table>
<p><a class="anchor" id="som"></a> </p>
<h2>SomaticMutation </h2>
<p>The <b>SomaticMutation</b> group contains a single data set <b>SomCall</b>.</p>
<p>The <b>SomCall</b> data set will contain the following columns:</p>
<table class="doxtable">
<tr>
<th>Column Name </th><th>Column Type </th><th>Description  </th></tr>
<tr>
<td>ProbeSetName </td><td>Ascii </td><td>The name of the probeset. </td></tr>
<tr>
<td>MutScore </td><td>Float </td><td>Mutation score </td></tr>
<tr>
<td>MutCall </td><td>Ascii </td><td>“Undetected”, “Low Confidence” or “High Confidence” </td></tr>
<tr>
<td>QNSignal </td><td>Float </td><td>Quantile normalized summary signal of relevant channel </td></tr>
<tr>
<td>MutThreshLow </td><td>Float </td><td>Minimum threshold for High Confidence MutCall </td></tr>
<tr>
<td>MutThreshHigh </td><td>Float </td><td>Minimum threshold for Low Confidence MutCall </td></tr>
</table>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated for Affymetrix Power Tools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
