<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.5"/>
<title>Affymetrix Power Tools: VIGNETTES: CytoScan750K Array</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<!-- end header part -->
<!-- Generated by Doxygen 1.8.5 -->
  <div id="navrow1" class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="LICENSE.html"><span>License</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div><!-- top -->
<div class="header">
  <div class="headertitle">
<div class="title"><h1>VIGNETTES: CytoScan750K Array </h1></div>  </div>
</div><!--header-->
<div class="contents">
<div class="textblock"><dl class="section date"><dt>Date</dt><dd>2015-03-03</dd></dl>
<h1><a class="anchor" id="content"></a>
Contents</h1>
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#libraryfiles">Library Files</a></li>
<li><a href="#ref">Command line example for reference creation:</a></li>
<li><a href="#ssa">Command line example for single sample mode processing</a></li>
</ul>
<p><a class="anchor" id="intro"></a> </p>
<h1><a class="anchor" id="intro"></a>
Introduction</h1>
<p>This vignette describes the process for using <code>apt-copynumber-cyto-ssa</code> and <code>apt-copynumber=cyto-ref</code> for reference model creation, and to process CytoScan750K arrays in single sample mode.</p>
<p><code>apt-copynumber-cyto-ref</code> implements two a reference model creation workflow that uses as a a set of CEL files that are input to outputs one or more reference files including copynumber and SNP information. <code>apt-copynumber-cyto-ssa</code> implements a single-sample workflow that compares each CEL file to a pre-computed reference for analysis. For efficiency of computation the "single-sample workflow" operates on a set of input CEL files at a time, but the output for any CEL file is unaffected by any of the other CEL files.</p>
<p><a class="anchor" id="libraryfiles"></a> </p>
<h1><a class="anchor" id="libraryfiles"></a>
Library Files</h1>
<p>The following library files are required for analysis of CytoScan750K array data.</p>
<pre class="fragment">CytoScan750K_Array.na32.1.annot.db:     NetAffx Annotation database file. It is a SQLite 3.x database.

CytoScan750K_Array.chrXprobes:          An ASCII file that contains probe_id (1-based) of
                                        probes on chrX. Used for copy number probe chrX/Y ratio gender calling.

CytoScan750K_Array.chrYprobes:          An ASCII file that contains probe_id (1-based) of
                                        probes on chrY as column 1 and probeset_id as column 2. Probe_id is used for
                                        copy number probe chrX/Y ratio gender calling and probeset_id is used for
                                        hasY gender call and Y-Gender call.

CytoScan750K_Array.r1.qca:              An ASCII file that defines the parameters for the calculations of qc-call-rate
                                        and contrast-qc-rand that are used in the geno-qc analysis.

CytoScan750K_Array.r1.qcc:              An ASCII file that defines the probesets that are used by the geno-qc analysis

CytoScan750K_Array.cdf:                 Chip definition file (binary file).

CytoScan750K_Array.probe_tab:           An ASCII file that contains probe sequence information. (Reference creation only).

CytoScan750K_Array.hg19.refcovar:       An hdf5 format file that contains covariate information for each probeset, where
                                        the first column is the probeset name and subsequent column(s) are covariates in float.
                                        (Reference creation only).

CytoScan750K_Array.snplist.txt:         An ASCII file that contains SNP probeset_id that are used to compute processed SNPQC and Raw SNPQC.
                                        The first row of the file must be "probeset_id". (Single sample mode processing only).

CytoScan750K_Array.na32.1.v1.REF_MODEL: Affymetrix copy number reference generated from a set of hapmaps and normal blood samples (hdf5 format file).
</pre><p><a class="anchor" id="ref"></a> </p>
<h1><a class="anchor" id="ref"></a>
Command line example for reference creation:</h1>
<p>NOTE: This command needs to have the comments removed to be runnable, as they break the unix shell syntax. The comments and additional spaces are to clarify the options being used. The arguments which are joined with a "."s should be joined without spaces.</p>
<pre class="fragment">apt-copynumber-cyto-ref -v 4 \             #verbose output level in the log file ranging from 1 (least details) to 4 (most details)
        --cyto2 false \                        #true: Cytogenetics_Array; false: CytoScanHD_Array,CytoScan750K_Array
        --force false \                        #true: Disable various checks including chip types; false: otherwise.
        --keep-intermediate-data false \       #true: dump out intermediate data during apt processing; false: otherwise.
        --doDualNormalization true \           #true: quantile normalize CN and SNP separately; false: do not quantile normalize CN and SNP separately
        --adapter-type-normalization false \   #true: perform adapter type normalization; false: otherwise. Should be always set to "false" for CytoScan750K_Array as the adjustment has been incorporated in "signal-adjustment-covariates"
        --probe-file CytoScan750K_Array.probe_tab \ #specify the probe sequence file
        --run-geno-qc true \                   #true: Run the GenoQC engine; false: otherwise
        --qca-file CytoScan750K_Array.r1.qca \ #if run-geno-qc is set to be true, it is required to provide this file
        --qcc-file CytoScan750K_Array.r1.qcc \ #if run-geno-qc is set to be true, it is required to provide this file
        --text-output false \                  #true: Output data in ASCII text format in addition to calvin format; false: otherwise
        --cnchp-output false \                 #true: report CNCHP file(s); false: otherwise. Either cnchp-output or cychp-output has to be true
        --cychp-output true \                  #true: report CYCHP file(s); false: otherwise. Either cnchp-output or cychp-output has to be true
        --reference-output CnNewRef.REF_MODEL \   #specify name of the new copy number reference file with file extension "REF_MODEL"
        --cdf-file CytoScan750K_Array.cdf \    #specify the chp definition file
        --chrX-probes CytoScan750K_Array.chrXprobes \ #specify the chrX probe file
        --chrY-probes CytoScan750K_Array.chrYprobes \ #specify the chrY probe file
        --annotation-file CytoScan750K_Array.na32.1.annot.db \  #specify the NetAffx database file
        --male-gender-ratio-cutoff 1.5 \       #male gender call threshold for chrX/Y ratio gender calling
        --female-gender-ratio-cutoff 0.9 \     #female gender call threshold for chrX/Y ratio gender calling
        --xx-cutoff 0.65 \                     #lower limit of the gender call threshold for hasXX gender call
        --xx-cutoff-high 0.95 \                #upper limit of the gender call threshold for hasXX gender call
        --y-cutoff 0.58 \                      #gender call threshold for hasY gender call
        --temp-dir temp_dir \                  #specify the temporary directory for intermediate data from reference generation
        --out-dir output_dir \                 #specify the output directory
        --covariates-file  CytoScan750K_Array.hg19.refcovar \ #specify the covariate file that contains probeset-based covariate(s) that doesn't exist in the annotation db file
        --signal-adjustment-covariates covariate-signal-adjuster.\  #covariate-signal adjuster
                order=fragment-adapter-type,fragment-length.\       #the covariates to be used in the order in which they are to be applied
                bin-type=discrete,equally-populated.\               #ordered list corresponding to each covariate controls whether the covariate is treated directly as discrete variable, or as a continuous variable (equally-populated or equally-spaced)
                bin-count=0,100 \                                   #ordered list corresponding to each covariate controls how a continuous variable is discretized into bins. Must be 0 for discrete covariates
        --lr-adjustment-covariates covariate-lr-adjuster.\          #covariate log2 ratios adjuster
                order=SuperGC,median-signal,marker-class.\          #the covariates to be used in the order in which they are to be applied
                bin-type=discrete,equally-populated,discrete.\      #ordered list correspoinding to each covariate controls whether the covariate is treated directly as discrete variable, or as a continuous variable (equally-populated or equally-spaced)
                bin-count=0,100,0\                                  #ordered list corresponding to each covariate controls how a continuous variable is discretized into bins. Must be 0 for discrete covariates
                .iqr-scaling=on,on,on.\                             #ordered list corresponding to each covariate on scales IQR for each bin to be equal
                subtract-from-XY=on,off,on \                        #ordered list corresponding to each covariate. “on” applies covariate adjustment to ChrX,ChrY markers.
        --wave-correction-reference-method wave-correction-reference-method.\ #estimate waves using the reference samples
                trim=2.0.\                                          #trim parameter for absolute adjusted log2 ratios after each wave
                percentile=0.75.\                                   #percentile to use for each probeset in finding wave. E.g., 0.75 means 75-th percentile.
                wave-count=6.\                                      #number of waves to calculate
                demean=false \                                      #true: demean prior to finding each wave; false: otherwise
        --local-gc-background-correction-reference-method none \    #should be always set to "none" for CytoScan750K_Array
        --local-gc-background-intensity-adjustment-method none \    #should be always set to "none" for CytoScan750K_Array
        --image-correction-intensity-adjustment-method none \       #should be always set to "none" for CytoScan750K_Array
        --allele-peaks-adjustment-covariates covariate-allele-peaks-adjuster.\ #covariate allelic difference adjuster
                order=local-gc.\                                    #the covariates to be used in the order in which they are to be applied
                bin-type=equally-populated.\                        #ordered list corresponding to each covariate controls whether the covariate is treated directly as discrete variable, or as a continuous variable (equally-populated or equally-spaced)
                bin-count=100.\                                     #ordered list corresponding to each covariate controls how a continuous variable is discretized into bins. Must be 0 for discrete covariates
                coarse-allele-peak-adjustment=on.\                  #if it is 'on' then turn on coarse allelic peak adjustment, and 'off' otherwise
                coarse-allele-peak-adjustment-step=50.\             #step size for another density estimation across the genome
                coarse-allele-peak-adjustment-window=200.\          #number of SNPs used by the kernel density estimator
                coarse-allele-peak-adjustment-point-count=128.\     #number of output grids for the kernel density estimator
                coarse-allele-peak-adjustment-bandwidth=0.25.\      #adjustment to the data-adaptive bandwidth for the kernel density estimator. Must be 0 &lt; bandwidth &lt;= 1.
                coarse-allele-peak-adjustment-cutoff=0.05.\         #parameter to remove density peaks that are likely due to noise
                coarse-allele-peak-adjustment-clean-threshold=0.35.\#controls with noisy markers are removed from the visualization because they are indeterminately far from adjacent peaks.
                coarse-allele-peak-adjustment-outlier-trim=3.0.\    #trim input allelic differences to be 3 or -3 if they go beyond those limits.
                master-peaks-point-count=513.\                      #number of output grids for the overall density estimator
                master-peaks-bandwidth=0.45.\                       #adjustment to the data-adaptive bandwidth for the per-covariate bin density estimator. Must be 0 &lt; bandwidth &lt;= 1 .
                covariate-bin-peaks-point-count=257.\               #number of output grids for per-covariate bin density estimator
                covariate-bin-peaks-bandwidth=0.45.\                #adjustment to the data-adaptive bandwidth for the per-covariate bin density estimator. Must be 0 &lt; bandwidth &lt;= 1 .
                kernel-function-selection=gaussian \                #The kernel choice for the density estimator. Currently only accept Gaussian kernel.
        --cel-files CELFileList.txt                                 #an ASCII file that contains the full path of input CEL files for reference creation, where the first row of the file is "cel_files" and each subsequent row corresponds to each CEL file
</pre><p><a class="anchor" id="ssa"></a> </p>
<h1><a class="anchor" id="ssa"></a>
Command line example for single sample mode processing</h1>
<p><code>apt-copynumber-cyto-ssa</code> can be run with or without norma diploid (ND) normalization. In cases where most of the genome is expected to be diploid then the standard form, without ND normalization should be used. ND normalization can be used in cases where this assumption does not apply.</p>
<p>For analysis without ND normalization the command line would look like the following:</p>
<pre class="fragment">apt-copynumber-cyto-ssa
    --allele-peak-count 24 \
    --nearest-power-of-two 512 \
    --keep-intermediate-data false \
    --dual-channel-normalization false \
    --mangle-probeset-names false \
    --waviness-block-size 50 \
    --waviness-genomic-span 0 \
    --igender-male-threshold 1.5 \
    --igender-female-threshold 0.9 \
    --sig-gender-xx-cutoff 0.65 \
    --sig-gender-xx-cutoff-high 0.95 \
    --sig-gender-y-cutoff 0.58 \
    --sig-gender-reference-chromosome 2 \
    --force false \
    --text-output false \
    --oschp-output false \
    --cychp-output true \
    --set-analysis-name cy750k \
    --snp-qc-snp-list $LIBDIR/CytoScan750K_Array.snplist.txt \
    --x-probes-file $LIBDIR/CytoScan750K_Array.chrXprobes    \
    --y-probes-file $LIBDIR/CytoScan750K_Array.chrYprobes \
    --annotation-file $LIBDIR/CytoScan750K_Array.na33.annot.db  \
    --reference-file $refInput \
    --y-target -0.50 \
    --out-dir $OUTDIR \
    --log-file $OUTDIR/apt-copynumber-cy750k-ssa.log \
    --temp-dir $OUTDIR/temp \
    --alpha-cn-calibrate 0.564278 \
    --alpha-X-cn-calibrate 0.619453 \
    --alpha-Y-cn-calibrate 0.494620 \
    --beta-cn-calibrate 1 \
    --beta-X-cn-calibrate 1 \
    --beta-Y-cn-calibrate 1 \
    --brlmmp-HARD 3 \
    --brlmmp-SB 0.450000 \
    --brlmmp-CM 2 \
    --brlmmp-bins 100 \
    --brlmmp-mix 1 \
    --brlmmp-bic 2 \
    --brlmmp-CSepPen 0.0 \
    --brlmmp-CSepThr 16 \
    --brlmmp-lambda 1.0 \
    --brlmmp-wobble 0.05 \
    --brlmmp-copyqc 0.000000 \
    --brlmmp-copytype 0 \
    --brlmmp-ocean 0.0 \
    --brlmmp-clustertype 1 \
    --brlmmp-transform mva \
    --brlmmp-MS 0.05 \
    --log2-ratio-adjuster-image-node:enable=true \
    --hpf-mini-block-rows 8 \
    --hpf-mini-block-cols 8 \
    --hpf-global-smooth-weight 256 \
    --hpf-local-smooth-weight 64 \
    --hpf-converged 0.000100 \
    --log2-ratio-adjuster-wave-node:enable=true \
    --wave-bandwidth 101 \
    --wave-bin-count 25 \
    --wave-count 6 \
    --wave-smooth true \
    --median-autosome-median-normalization true \
    --ad-outlier-trim 3.0 \
    --ap-ad-step 20 \
    --ap-ad-window 100 \
    --ap-ad-point-count 129 \
    --ap-ad-bandwidth 0.25 \
    --ap-ad-cutoff 0.05 \
    --ap-ad-threshold 0.35 \
    --ap-ad-symmetry true \
    --ap-ad-height-threshold 0.000000 \
    --ap-ad-height-threshold-bound 1024.000000 \
    --normal-diploid-normalization false \
    --ndd-weights-type equal \
    --ndd-point-count 128 \
    --ndd-bandwidth 0.25 \
    --ndd-cutoff 0.05 \
    --ndd-step 40 \
    --ndd-window 275 \
    --ndd-min-markers 10 \
    --ndd-log2-threshold 0.28 \
    --ndd-th1 0,0.4 \
    --ndd-th2 0.7,1.3 \
    --ndd-th3 0.7,1.3 \
    --ndd-th4 0,0.4 \
    --ndd-min-percent-nd 0 \
    --ndd-min-number-nd 2000 \
    --ndd-max-number-nd 375000 \
    --ndd-med-l2r-low-percentile-ND 0.015 \
    --ndd-med-l2r-threshold-ND 0.20 \
    --ndd-iqr-factor-coarse 5.0 \
    --ndd-iqr-factor-fine 3.0 \
    --nddlrcov-min-nd-marker-count-per-bin 30 \
    --nddlrcov-use-nd-lr-markers on \
    --nddqc-min-count 2000 \
    --nddicov-min-nd-probe-count-per-bin 30 \
    --nddicov-use-nd-signal-markers off \
    --kernel-sigma-span 50 \
    --hmmCN_state 0,1,2,3,4 \
    --hmmCN_mu -2,-0.5,0,0.33,0.54 \
    --hmmCN_sigma 0.3,0.3,0.3,0.3,0.3 \
    --hmmCN_state-X 0,1,2,3,4 \
    --hmmCN_mu-X -2,-0.55,0,0.36,0.57 \
    --hmmCN_sigma-X 0.3,0.3,0.3,0.3,0.3 \
    --hmmCN_state-Y 0,1,2,3,4 \
    --hmmCN_mu-Y -1.4,-0.5,0,0.3,0.51 \
    --hmmCN_sigma-Y 0.3,0.3,0.3,0.3,0.3 \
    --diagonal-weight-Y 0.995 \
    --mapd-weight-Y 0.2 \
    --min-segment-size-Y 5 \
    --hmm-confidence-weight-Y 0.6 \
    --diagonal-weight 0.995 \
    --mapd-weight 0.2 \
    --min-segment-size 5 \
    --hmm-confidence-weight 0.6 \
    --hmm-shrink true \
    --diagonal-weight-X 0.995 \
    --mapd-weight-X 0.2 \
    --min-segment-size-X 5 \
    --hmm-confidence-weight-X 0.6 \
    --hmm-shrink true \
    --agr-denominator-source CN \
    --agr-denominator-percentile 0.5 \
    --cn-gender-cutoff 0.5 \
    --loh-error-rate 0.05 \
    --loh-beta 0.001 \
    --loh-alpha 0.01 \
    --loh-separation 1000000 \
    --loh-min-marker-count 10 \
    --loh-no-call-threshold 0.05 \
    --loh-min-genomic-span 3000000 \
    --gains-boundries 0.0820,0.1588,0.2318,0.3130 \
    --losses-boundries -0.1049,-0.2263,-0.3477,-0.4691 \
    --marker-bandwidth 6000 \
    --confidence-window 251 \
    --run-y-chromosome false \
    --cel-files $CELFileList \
    --cn-neutral-loh-node:enable=true
</pre><p>For analysis without ND normalization the command line would look like the following:</p>
<pre class="fragment">apt-copynumber-cyto-ssa

    --allele-peak-count 24 \
    --nearest-power-of-two 512 \
    --keep-intermediate-data false \
    --dual-channel-normalization false \
    --mangle-probeset-names false \
    --waviness-block-size 50 \
    --waviness-genomic-span 0 \
    --igender-male-threshold 1.5 \
    --igender-female-threshold 0.9 \
    --sig-gender-xx-cutoff 0.65 \
    --sig-gender-xx-cutoff-high 0.95 \
    --sig-gender-y-cutoff 0.58 \
    --sig-gender-reference-chromosome 2 \
    --force false \
    --text-output false \
    --oschp-output false \
    --cychp-output true \
    --set-analysis-name cy750k \
    --snp-qc-snp-list $LIBDIR/CytoScan750K_Array.snplist.txt \
    --x-probes-file $LIBDIR/CytoScan750K_Array.chrXprobes    \
    --y-probes-file $LIBDIR/CytoScan750K_Array.chrYprobes \
    --annotation-file $LIBDIR/CytoScan750K_Array.na33.annot.db  \
    --reference-file $refInput \
    --y-target -0.50 \
    --out-dir $OUTDIR \
    --log-file $OUTDIR/apt-copynumber-cy750k-ssa.log \
    --temp-dir $OUTDIR/temp \
    --alpha-cn-calibrate 0.564278 \
    --alpha-X-cn-calibrate 0.619453 \
    --alpha-Y-cn-calibrate 0.494620 \
    --beta-cn-calibrate 1 \
    --beta-X-cn-calibrate 1 \
    --beta-Y-cn-calibrate 1 \
    --brlmmp-HARD 3 \
    --brlmmp-SB 0.450000 \
    --brlmmp-CM 2 \
    --brlmmp-bins 100 \
    --brlmmp-mix 1 \
    --brlmmp-bic 2 \
    --brlmmp-CSepPen 0.0 \
    --brlmmp-CSepThr 16 \
    --brlmmp-lambda 1.0 \
    --brlmmp-wobble 0.05 \
    --brlmmp-copyqc 0.000000 \
    --brlmmp-copytype 0 \
    --brlmmp-ocean 0.0 \
    --brlmmp-clustertype 1 \
    --brlmmp-transform mva \
    --brlmmp-MS 0.05 \
    --log2-ratio-adjuster-image-node:enable=false \
    --hpf-mini-block-rows 8 \
    --hpf-mini-block-cols 8 \
    --hpf-global-smooth-weight 256 \
    --hpf-local-smooth-weight 64 \
    --hpf-converged 0.000100 \
    --log2-ratio-adjuster-wave-node:enable=true \
    --wave-bandwidth 101 \
    --wave-bin-count 25 \
    --wave-count 6 \
    --wave-smooth true \
    --median-autosome-median-normalization true \
    --ad-outlier-trim 3.0 \
    --ap-ad-step 20 \
    --ap-ad-window 100 \
    --ap-ad-point-count 129 \
    --ap-ad-bandwidth 0.25 \
    --ap-ad-cutoff 0.05 \
    --ap-ad-threshold 0.35 \
    --ap-ad-symmetry true \
    --ap-ad-height-threshold 0.000000 \
    --ap-ad-height-threshold-bound 1024.000000 \
    --normal-diploid-normalization true \
    --fld-file $LIBDIR/CytoScan750K_Array.NDN.FLD.r1.txt \
    --ndd-weights-type FLD \
    --ndd-point-count 128 \
    --ndd-bandwidth 0.25 \
    --ndd-cutoff 0.05 \
    --ndd-step 40 \
    --ndd-window 275 \
    --ndd-min-markers 10 \
    --ndd-log2-threshold 0.28 \
    --ndd-th1 0,0.4 \
    --ndd-th2 0.7,1.3 \
    --ndd-th3 0.7,1.3 \
    --ndd-th4 0,0.4 \
    --ndd-min-percent-nd 0 \
    --ndd-min-number-nd 2000 \
    --ndd-max-number-nd 375000 \
    --ndd-med-l2r-low-percentile-ND 0.015 \
    --ndd-med-l2r-threshold-ND 0.20 \
    --ndd-iqr-factor-coarse 5.0 \
    --ndd-iqr-factor-fine 3.0 \
    --nddlrcov-min-nd-marker-count-per-bin 30 \
    --nddlrcov-use-nd-lr-markers on \
    --nddqc-min-count 2000 \
    --nddicov-min-nd-probe-count-per-bin 30 \
    --nddicov-use-nd-signal-markers off \
    --kernel-sigma-span 50 \
    --hmmCN_state 0,1,2,3,4 \
    --hmmCN_mu -2,-0.5,0,0.33,0.54 \
    --hmmCN_sigma 0.3,0.3,0.3,0.3,0.3 \
    --hmmCN_state-X 0,1,2,3,4 \
    --hmmCN_mu-X -2,-0.55,0,0.36,0.57 \
    --hmmCN_sigma-X 0.3,0.3,0.3,0.3,0.3 \
    --hmmCN_state-Y 0,1,2,3,4 \
    --hmmCN_mu-Y -1.4,-0.5,0,0.3,0.51 \
    --hmmCN_sigma-Y 0.3,0.3,0.3,0.3,0.3 \
    --diagonal-weight-Y 0.995 \
    --mapd-weight-Y 0.2 \
    --min-segment-size-Y 5 \
    --hmm-confidence-weight-Y 0.6 \
    --diagonal-weight 0.995 \
    --mapd-weight 0.2 \
    --min-segment-size 5 \
    --hmm-confidence-weight 0.6 \
    --hmm-shrink true \
    --diagonal-weight-X 0.995 \
    --mapd-weight-X 0.2 \
    --min-segment-size-X 5 \
    --hmm-confidence-weight-X 0.6 \
    --hmm-shrink true \
    --agr-denominator-source CN \
    --agr-denominator-percentile 0.5 \
    --cn-gender-cutoff 0.5 \
    --loh-error-rate 0.05 \
    --loh-beta 0.001 \
    --loh-alpha 0.01 \
    --loh-separation 1000000 \
    --loh-min-marker-count 10 \
    --loh-no-call-threshold 0.05 \
    --loh-min-genomic-span 3000000 \
    --gains-boundries 0.0820,0.1588,0.2318,0.3130 \
    --losses-boundries -0.1049,-0.2263,-0.3477,-0.4691 \
    --marker-bandwidth 6000 \
    --confidence-window 251 \
    --run-y-chromosome false \
    --cel-files $CELFileList \
    --cn-neutral-loh-node:enable=true</pre> </div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated for Affymetrix Power Tools by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.5
</small></address>
</body>
</html>
