<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: VIGNETTES: Mouse Diversity Genotyping Array Clustering Analysis</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="LICENSE.html"><span>License</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>VIGNETTES: Mouse Diversity Genotyping Array Clustering Analysis </h1>  </div>
</div>
<div class="contents">
<dl class="date"><dt><b>Date:</b></dt><dd>2009-6-15</dd></dl>
<h2><a class="anchor" id="content"></a>
Contents</h2>
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#supporting">Supporting Files</a></li>
<li><a href="#apt">APT Installation</a></li>
<li><a href="#sample">Sample Genotyping Script</a></li>
<li><a href="#output">Output</a></li>
</ul>
<h2><a class="anchor" id="intro"></a>
Introduction</h2>
<p>The Mouse Diversity Genotyping Array is designed for high-density, genome-wide profiling of single nucleotide polymorphisms (SNPs) and copy number variation (CNV) segments. This cutting-edge research tool provides more than 100 times the SNP coverage than any other available mouse array, permitting high resolution mapping, genomic analysis, association studies, etc.</p>
<p>This vignette provides examples on how to use Affymetrix Power Tools (APT) to make genotype calls from the array. These examples and the supporting files are designed for genotyping SNPs in classic mouse strains.</p>
<p>Before genotyping, it is important to remove the samples that do not pass QC. Please see the <a href="VIGNETTE-Mouse-WGSA-sampleqc.html">Mouse Diversity Genotyping Array Sample Quality Control Vignette</a>.</p>
<h2><a class="anchor" id="supporting"></a>
Supporting Files</h2>
<p>The following files are referenced in this vignette:</p>
<ul>
<li>Sample Genotyping Script</li>
<li>CDF Library file</li>
<li>Sample CEL files</li>
<li>Sample CEL file list</li>
<li>SNP subset file</li>
<li>Models file</li>
<li>Chromosome X Probe file</li>
<li>Chromosome Y Probe file</li>
<li>Special SNPs file</li>
<li>Sample output files</li>
</ul>
<p>These files can be found on the Affymetrix Mouse Diversity Genotyping Array <a href="http://www.affymetrix.com/estore/browse/products.jsp?productId=prod100002&amp;categoryId=cat30008&amp;productName=Affymetrix%26%23174%3B-Mouse-Diversity-Genotyping-Array#1_3">product page</a>.</p>
<h2><a class="anchor" id="apt"></a>
APT Installation</h2>
<p>Affymetrix Power Tools (APT) are a set of cross-platform command line programs that implement algorithms for analyzing and working with Affymetrix GeneChip arrays. The latest version of the Affymetrix Power Tools can be downloaded and installed from here: <a href="http://www.affymetrix.com/partners_programs/programs/developer/tools/powertools.affx#1_2">http://www.affymetrix.com/partners_programs/programs/developer/tools/powertools.affx#1_2</a></p>
<p>The program for this vignette, "apt-probeset-genotype", will be in "&lt;apt folder&gt;\bin" after downloading and unzipping.</p>
<h2><a class="anchor" id="sample"></a>
Sample Genotyping Script</h2>
<p>apt-probeset-genotype is the application for making genotype calls. A modified version of BRLMM-P is the genotype calling algorithm used in this array. It is a model based algorithm, so it needs to be run on multiple CEL files at once to estimate probe effect and SNP cluster parameters. It is advisable to cluster with at least 44 genetically distinct samples, though adding more will continue to be of benefit, in particular for accurately calling rare genotypes.</p>
<p>Included in the files available on the array <a href="http://www.affymetrix.com/estore/browse/products.jsp?productId=prod100002&amp;categoryId=cat30008&amp;productName=Affymetrix%26%23174%3B-Mouse-Diversity-Genotyping-Array#1_3">product page</a> and included below is a sample genotyping script "genotype.sh". To execute this script without change, "genotype.sh", "apt-probeset-genotype", and all input files must be in the same directory.</p>
<p>The command below is very similar to that used for doing single-sample analysis but there is one key difference. The parameters supplied in the analysis string specify a multi-sample clustering mode. In this mode, data from all samples is used in a Bayesian process to update the prior estimates of genotype cluster locations before making final genotype calls.</p>
<div class="fragment"><pre class="fragment">
  apt-probeset-genotype \
	-o genotype \
	-c MOUSEDIVm520650.CDF \
	--cel-files cel.txt \
	-s SNPsubset.txt \
	--a quant-norm.sketch=50000,pm-only,brlmm-p.CM=1.bins=100.mix=1.bic=2.HARD=3.SB=0.75.KX=0.2.KH=0.3.KXX=-0.1.KAH=-0.1.KHB=-0.1.transform=MVA.AAM=2.0.BBM=-2.0.AAV=0.06.BBV=0.06.ABV=0.06.copyqc=0.00000.wobble=0.05.CSepThr=4.CSepPen=0.1.KYAH=-0.05.KYHB=-0.05.KYAB=-0.1.AAY=9.ABY=9.5.BBY=9.copytype=-1.clustertype=2.ocean=0.00001.MS=0.1 \
	--read-models-brlmmp Mouse.models \
	--summaries \
	--write-models
	--chrX-probes Mouse.chrXprobes \
	--chrY-probes Mouse.chrYprobes \
	--special-snps Mouse.specialSNPs
</pre></div><ul>
<li>-o Identifies the output directory.</li>
<li>-c Identifies the CDF library file.</li>
<li>--cel-files Identifies a text file with column 'cel_files' specifying the location of the files to process.<ul>
<li>Although the algorithm is currently tuned for best performance on classic inbred mouse strains, the sample CEL files made available on the array <a href="http://www.affymetrix.com/estore/browse/products.jsp?productId=prod100002&amp;categoryId=cat30008&amp;productName=Affymetrix%26%23174%3B-Mouse-Diversity-Genotyping-Array#1_3">product page</a> include inbred and F1 samples. These samples are available to test the above genotyping script. They may also be of use for clustering with other customer samples.</li>
</ul>
</li>
<li>-s Identifies a text file with column 'probeset_id' specifying probesets (SNPs) to genotype.<ul>
<li>The 584,729 SNPs in the subset file that accompanies this vignette are designed for optimum performance in classic mouse strains. This parameter can be removed to genotype all SNPs on the array.</li>
</ul>
</li>
<li>--a String representing analysis pathway desired.<ul>
<li>The "MS=0.1" specification in the analysis string sets the confidence threshold for no-calls. The value must be between 0 and 1, with higher values corresponding to lower stringency (enabling more calls though at less accuracy). Values other than the default should be used with a suitable degree of caution.</li>
</ul>
</li>
<li>--read-models-brlmmp Identifies a models file consisting of SNP-specific priors specifying where the 3 genotype clusters are likely to be located for each SNP.</li>
<li>--summaries Output the summary values for each allele. Useful for making cluster plots of SNPs of interest.</li>
<li>--write-models Output the SNP Specific models.</li>
<li>--chrX-probes File containing chrX probe_ids. Used for probe chrY/X ratio gender calling.</li>
<li>--chrY-probes File containing chrY probe_ids. Used for probe chrY/X ratio gender calling.</li>
<li>--special-snps File consisting of the probeset_ids for all SNPs with non-diploid copy number.</li>
</ul>
<p><em>Note</em> - In unix, there should not be any characters after the continuation of a command '\'.</p>
<p><em>Note</em> - The windows DOS prompt does not allow a continuation of a command with the '\' character. So in the sample script above, the '\' character should be omitted and everything entered on a single line.</p>
<p>For more information and options for the genotyping script, please see the apt-probeset-genotype manual: <a href="http://www.affymetrix.com/support/developer/powertools/changelog/apt-probeset-genotype.html">http://www.affymetrix.com/support/developer/powertools/changelog/apt-probeset-genotype.html</a></p>
<p>For more information on the genotyping algorithm, see the original <a href="http://www.affymetrix.com/support/technical/whitepapers/brlmmp_whitepaper.pdf">BRLMM-P whitepaper</a> as well as the <a href="http://www.affymetrix.com/support/technical/whitepapers/dmet_plus_algorithm_whitepaperv1.pdf">DMET Plus algorithm whitepaper</a> which documents some more recently-added features.</p>
<h2><a class="anchor" id="output"></a>
The output report</h2>
<p>The following is the standard genotyping output. All, except for the summary files, are included with the download and will be output in the genotype directory when "genotype.sh" is executed.</p>
<ul>
<li>quant-norm.pm-only.brlmm-p.calls.txt<ul>
<li>Tab-delimited text file containing the genotype calls for the SNPs analyzed. Each column corresponds with a sample (CEL file). Each row corresponds with a SNP. The calls can be the following:<ul>
<li>0=AA: SNP is homozygous for the reference (B6) allele.</li>
<li>1=AB: SNP is heterozygous.</li>
<li>2=BB: SNP is homozygous for the variant allele.</li>
<li>-1=N: SNP was not called due to low confidence.</li>
</ul>
</li>
</ul>
</li>
<li>quant-norm.pm-only.brlmm-p.confidences.txt<ul>
<li>Tab-delimited text file containing the confidence values for the genotype calls. This is a value between 0 and 1 that measures the quality of the call. The lower the value, the greater the confidence in the call. The confidence threshold is set to 0.1 in the sample genotyping script. Calls with a larger confidence value are marked as not called. This can be adjusted in the analysis string ("MS" parameter) to trade off between call rate and accuracy.</li>
</ul>
</li>
<li>quant-norm.pm-only.brlmm-p.report.txt<ul>
<li>A tab-delimited text file containing a summary report with a quick overview of the samples (CELs files) analyzed.</li>
</ul>
</li>
<li>quant-norm.pm-only.brlmm-p.summary.txt<ul>
<li>A tab-delimited text file containing summary values in contrast space used in the clustering. This is useful for drawing SNP cluster plots in A vs B space.</li>
</ul>
</li>
<li>quant-norm.pm-only.brlmm-p.normalized-summary.txt<ul>
<li>A tab-delimited text file containing the normalized and transformed summary values that are passed in to the genotyping algorithm. This is useful for drawing SNP cluster plots in the space in which BRLMM-P does its clustering.</li>
</ul>
</li>
<li>quant-norm.pm-only.brlmm-p.snp-posteriors.txt<ul>
<li>This is useful for getting a sense of the final SNP posteriors that were determined from the input data and the SNP priors, which in turn can help guide intuition as to how the final genotype calls are made.</li>
</ul>
</li>
</ul>
<p>For more information on the output files, please see the apt-probeset-genotype manual: <a href="http://www.affymetrix.com/support/developer/powertools/changelog/apt-probeset-genotype.html">http://www.affymetrix.com/support/developer/powertools/changelog/apt-probeset-genotype.html</a></p>
<p><b></p>
<pre>Affymetrix Power Tools (APT) Release 1.18.0</pre><p></b> </p>
</div>
<hr class="footer"/><address class="footer"><small>Generated for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
