<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: VIGNETTES: Single-sample Genotype Calling with the Whole Genome Sampling Assay</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="LICENSE.html"><span>License</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>VIGNETTES: Single-sample Genotype Calling with the Whole Genome Sampling Assay </h1>  </div>
</div>
<div class="contents">
<dl class="date"><dt><b>Date:</b></dt><dd>2008-06-22</dd></dl>
<p><b>This is an experimental workflow for generating genotype calls one sample at a time. Do your own due diligence in using this workflow and interpreting the results. One should not use this for final and optimal genotype calling, but rather for applications such as quality assurance including sample tracking where a single sample approach may be desirable from a workflow perspective. Again, DO NOT use these single sample genotype calls for your association study.</b></p>
<h2><a class="anchor" id="content"></a>
Contents</h2>
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#training">Training the required parameters</a></li>
<li><a href="#using">Genotype calling</a></li>
</ul>
<h2><a class="anchor" id="intro"></a>
Introduction</h2>
<p>Accuarate determination of SNP genotypes is best accomplished by clustering together samples in batches. The availability of multiple samples processed at the same time and under similar circumstances enables genotype clustering algorithms to estimate the positions and variances of genotype clusters - this is particularly important when the clusters are positioned away from the average locations.</p>
<p>That said, there are still situations where it is useful to call genotypes based on a single sample. Examples include situations where very few samples were processed, or when one wants to do a check of sample identity via a quick check on genotypes without going through a full clustering. When using the single-sample analysis workflow it is important to bear in mind the risk that the accuracy of the calls may be lower, perhaps substantially, than what is attainable via the recommended multiple sample clustering.</p>
<p>The single-sample workflow described here uses the BRLMM-P genotyping algorithm, which has two key enabling features - it has the capability to enable the user to generate a set of SNP-specific priors and it has a special single-samples calling mode.</p>
<h2><a class="anchor" id="training"></a>
Training the required parameters</h2>
<p>Single-sample genotype calling requries three things</p>
<ul>
<li>A target probe intensity distribution to which to normalize</li>
<li>Estimated probe effects to use when generating signal estimates for each allele</li>
<li>Priors on the locations, variance and frequence of the three genotype clusters</li>
</ul>
<p>The target probe intensity distribution can be created with the --write-sketch option to apt-probeset-genotype - this option causes the quantile sketch derived to be written out to a file with filename suffix ".normalization-target.txt" in the output directory. Estimates of probe (feature) effects can also be generated with apt-probeset-genotype using the ----feat-effects option - the resulting feature effects are written to a file with suffix ".feature-response.txt" in the output directory.</p>
<p>To generate a set of SNP-specific priors, see the <a href="VIGNETTE-WGSA-clustering-without-priors.html">clustering without priors and building priors</a>. Even if a set of priors is already available (for example, if you are using one of the catalog products) it may be useful to allow your own samples influence the priors by starting with the set of provided priors, doing a clustering run with your own samples, then saving the resulting posteriors to serve as priors for future runs. This can be accomplished with the --write-models option to apt-probeset-genotype.</p>
<p>An example of using the above three options in a run using the GenomeWideSNP_6 product is shown below.</p>
<div class="fragment"><pre class="fragment">
apt-probeset-genotype \
  --cdf-file             lib/GenomeWideSNP_6.cdf \
  --special-snps         lib/GenomeWideSNP_6.specialSNPs \
  --read-models-brlmmp   lib/GenomeWideSNP_6.brlmm-p.models \
  --chrX-probes          lib/GenomeWideSNP_6.chrXprobes \
  --chrY-probes          lib/GenomeWideSNP_6.chrYprobes \
  --set-gender-method    cn-probe-chrXY-ratio \
  --analysis             brlmm-p-plus \
  --out-dir              single_sample_parameter_dir \
  --write-models \
  --write-sketch \
  --summaries \
  --feat-effects \
  --feat-eff-remove-allele-suffix \
  --cel-files            my_cel_file_list.txt
</pre></div><p>Note that the analysis string should match that which was used for the models file supplied. In the case of the GenomeWideSNP_6 product there is a models file GenomeWideSNP_6.brlmm-p.models supplied as part of the library file package found on the <a href="http://www.affymetrix.com/products/arrays/specific/genome_wide_snp6/genome_wide_snp_6.affx">product page</a>, it was created with the same analysis string that is aliased to "brlmm-p-plus".</p>
<h2><a class="anchor" id="using"></a>
Genotype calling</h2>
<p>Following on from the example of how to train the required parameters, here is an example of how to use them to perform single-sample genotype calling:</p>
<div class="fragment"><pre class="fragment">
apt-probeset-genotype \
  --cdf-file             lib/GenomeWideSNP_6.cdf \
  --special-snps         lib/GenomeWideSNP_6.specialSNPs \
  --chrX-probes          lib/GenomeWideSNP_6.chrXprobes \
  --chrY-probes          lib/GenomeWideSNP_6.chrYprobes \
  --set-gender-method    cn-probe-chrXY-ratio \
  --analysis             quant-norm.sketch=50000,pm-only,brlmm-p.CM=2.bins=100.mix=1.bic=2.HARD=3.SB=0.45.KX=1.KH=1.5.KXX=0.5.KAH=-0.6.KHB=-0.6.transform=MVA.AAM=2.0.BBM=-2.0.AAV=0.06.BBV=0.06.ABV=0.06.copyqc=0.000001.wobble=0.05.MS=0.05 \
  --out-dir              my_output_dir \
  --use-feat-eff         single_sample_parameter_dir/brlmm-p-plus.plier.feature-response.txt  \
  --target-sketch        single_sample_parameter_dir/quant-norm.normalization-target.txt  \
  --read-models-brlmmp   single_sample_parameter_dir/brlmm-p-plus.snp-posteriors.txt \
  --cel-files            my_cel_file_list.txt
</pre></div><p>Note that the long analysis string is the same as the analysis string aliased to "brlmm-p-plus" as used in training the SNP-specific priors, but with the critical distinction that "CM=2". CM stands for Calling Method and setting it to 2 enables the single-sample calling mode.</p>
<p>As a test you can analyze a collection of CEL files, save the feature effects, quantile sketch and SNP-specific priors, and then recall all the samples in single-sample mode. This should result in the same genotype calls for both modes of calling, with some small amount of difference due to the rounding involved when the parameters are written out to text files.</p>
<p><b></p>
<pre>Affymetrix Power Tools (APT) Release 1.18.0</pre><p></b> </p>
</div>
<hr class="footer"/><address class="footer"><small>Generated for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
