<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<title>Affymetrix Power Tools: VIGNETTES: Use of APT to Analyze DMET Plus Arrays</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<link href="doxygen.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<!-- Generated by Doxygen 1.7.1 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul class="tablist">
             <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
             <li><a href="CHANGE.html"><span>Change&nbsp;Log</span></a></li>
             <li><a href="VIGNETTE.html"><span>Vignettes</span></a></li>
             <li><a href="FAQ.html"><span>FAQs</span></a></li>
             <li><a href="LICENSE.html"><span>License</span></a></li>
             <li><a href="FILE-FORMATS.html"><span>File Formats</span></a></li>
             <li><a href="PLATFORMS.html"><span>Platforms</span></a></li>
    </ul>
  </div>
</div>
<div class="header">
  <div class="headertitle">
<h1>VIGNETTES: Use of APT to Analyze DMET Plus Arrays </h1>  </div>
</div>
<div class="contents">
<dl class="date"><dt><b>Date:</b></dt><dd>2011-04-07</dd></dl>
<h2><a class="anchor" id="content"></a>
Contents</h2>
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#fixed">Fixed boundary analysis</a></li>
<li><a href="#dyn">Dynamic boundary analysis</a></li>
<li><a href="#access">Accessing the genotyping and copy number results</a></li>
</ul>
<h2><a class="anchor" id="intro"></a>
Introduction</h2>
<p>Affymetrix Power Tools includes command line access to the computational engines used for the DMET genotyping and copy number analysis in the DMET Console application. This access provides both flexibility to incorporate the computational engine into custom bioinformatic pipelines and also exposes certain options not currently available through DMET Console. This vignette outlines some examples of command line processing of DMET data using different available modes of analysis.</p>
<p>You will require the <a href="http://www.affymetrix.com/Auth/support/downloads/library_files/dmet_plus_genotyping_libraryfile.zip">DMET_Plus genotyping library files</a> and the <a href="http://www.affymetrix.com/Auth/support/downloads/library_files/dmet_plus_annot_translation.zip">DMET Plus annotation and translation files</a>. The library files include all supported versions of the input reference file for the DMET Plus array. In addition, you need version 1.12.0 or later of the <a href="http://www.affymetrix.com/partners_programs/programs/developer/tools/powertools.affx#1_1">Affymetrix Power Tools</a>. The CHP files generated by the workflows presented here can be incorporated into DMET Console for further analysis and visualization.</p>
<p>The DMET_Plus genotyping library files are assumed to reside in a directory indicated by the path stored in $LIB_DIR. The DMET Plus annotation and translation files are assumed to reside in a directory indicated by the path stored in $ANNOT_DIR.</p>
<p>Most of the library files remain the same moving among DMET Console versions 1.0, 1.1, and 1.2; only the genomic input reference and plasmid input reference files are revised. The different reference files include the following:</p>
<table  border="1" cellspacing="0">
<tr>
<th>File </th><th>DNA Type </th><th>Fixed/Dynamic </th><th>Version  </th></tr>
<tr>
<td>DMET_Plus.v1.genomic.ref.a5 </td><td>Genomic </td><td>Fixed </td><td>1  </td></tr>
<tr>
<td>DMET_Plus.v1.genomic.ref.r2.a5 </td><td>Genomic </td><td>Dynamic </td><td>1  </td></tr>
<tr>
<td>DMET_Plus.v1.genomic.ref.r3.a5 </td><td>Genomic </td><td>Fixed </td><td>2  </td></tr>
<tr>
<td>DMET_Plus.v1.genomic.ref.r4.a5 </td><td>Genomic </td><td>Dynamic </td><td>2  </td></tr>
<tr>
<td>DMET_Plus.v1.plasmid.ref.a5 </td><td>Plasmid </td><td>Fixed </td><td>1  </td></tr>
<tr>
<td>DMET_Plus.v1.plasmid.ref.r2.a5 </td><td>Plasmid </td><td>Dynamic </td><td>1  </td></tr>
</table>
<p>The general advice is to use the newest versions of the reference files, either fixed boundary or dynamic boundary as desired. The original reference files are provided principally for compatibility with previous work.</p>
<h2><a class="anchor" id="fixed"></a>
Fixed boundary analysis</h2>
<p>The following call to apt-dmet-genotype processes the CEL files in the same manner as the legacy "Fixed Genotype Boundaries" analysis configuration for DMET Console. Other DMET Plus input reference files will function properly for single-sample analysis, but they will not yield numerical results that match those generated by DMET Console. In fixed-boundary analysis, samples are analyzed one-at-a-time and signals are compared against predefined clustering models to make a genotype call. There are separate clustering models for plasmid controls and genomic samples, so make sure to match the right models with the right sample type.</p>
<div class="fragment"><pre class="fragment">
  apt-dmet-genotype \
    --sample-type                genomic \
    --out-dir                    output \
    --cdf-file                   $LIB_DIR/DMET_Plus.v1.cdf \
    --cel-files                  cel_files.txt \
    --run-cn-engine              true \
    --chrX-probes                $LIB_DIR/DMET_Plus.v1.chrXprobes \
    --chrY-probes                $LIB_DIR/DMET_Plus.v1.chrYprobes \
    --special-snps               $LIB_DIR/DMET_Plus.v1.specialSNPs \
    --region-model               $LIB_DIR/DMET_Plus.v1.cn-region-models.txt \
    --probeset-model             $LIB_DIR/DMET_Plus.v1.cn-probeset-models.txt \
    --reference-input            $LIB_DIR/DMET_Plus.v1.genomic.ref.a5 \
    --gt-analysis                quant-norm.sketch=50000,pm-only,brlmm-p-multi\
.CM=2.bins=100.mix=1.bic=2.lambda=0.0.HARD=3.SB=0.75.KX=0.3.KH=0.3.KXX=0.1\
.KAH=-0.1.KHB=-0.1.KYAH=-0.05.KYHB=-0.05.KYAB=-0.1.transform=MVA.AAM=2.8\
.BBM=-2.8.AAV=0.10.BBV=0.10.ABV=0.10.V=1.AAY=10.7.ABY=11.3.BBY=10.7\
.copyqc=0.00000.wobble=0.0.MS=0.1.copytype=-1.clustertype=2.CSepPen=0.5\
.ocean=0.0000001.cc-alleles=6.cc-type=UCHAR.cc-version=1.0 \
    --geno-call-thresh           0.1 \
    --cc-chp-output              true \
    --probeset-ids               $LIB_DIR/DMET_Plus.v1.genomic.gt.ps \
    --cn-region-gt-probeset-file $LIB_DIR/DMET_Plus.v1.cn-gt.ps \
    --chip-type                  DMET_Plus
</pre></div><ul>
<li>The '--sample-type' option indicates whether genomic or plasmid samples are to be processed. The '--reference-input' and '--probeset-ids' options should select a consistent library files.</li>
</ul>
<ul>
<li>The '--out-dir' option specifies the output directory and the '--cel-files' option identifies a file containing a list of CEL files to be processed.</li>
</ul>
<ul>
<li>The '--run-cn-engine' option determines whether to perform copy number analysis.</li>
</ul>
<ul>
<li>The '--cc-chp-output' option controls whether the CHP files are output. This option should be set to true in order to generate the CHP files needed for subsequent processing.</li>
</ul>
<ul>
<li>The '--geno-call-thresh' option controls the confidence level at which genotypes calls are made, and the value 0.1 represents the default.</li>
</ul>
<ul>
<li>The '--gt-analysis' option contains the algorithm string that controls the clustering behavior. The detailed meaning of these different algorithmic parameters is explained in the output from the following call:</li>
</ul>
<div class="fragment"><pre class="fragment">
  apt-probeset-genotype --explain brlmm-p
</pre></div><p>Examine the documentation for <a href="http://media.affymetrix.com/support/developer/powertools/changelog/apt-dmet-genotype.html">apt-dmet-genotype</a> for a detailed explanation of all available options and their meanings.</p>
<p>To process plasmids, set --sample-type to plasmid, --reference-input to DMET_Plus.v1.plasmid.ref.a5, and --probeset-ids to DMET_Plus.v1.plasmid.gt.ps, respectively.</p>
<p>The following call to apt-dmet-genotype processes the CEL files in the same manner as the "Fixed Genotype Boundaries &mdash; version 2" analysis configuration for DMET Console.</p>
<div class="fragment"><pre class="fragment">
  apt-dmet-genotype \
    --sample-type                genomic \
    --out-dir                    output \
    --cdf-file                   $LIB_DIR/DMET_Plus.v1.cdf \
    --cel-files                  cel_files.txt \
    --run-cn-engine              true \
    --chrX-probes                $LIB_DIR/DMET_Plus.v1.chrXprobes \
    --chrY-probes                $LIB_DIR/DMET_Plus.v1.chrYprobes \
    --special-snps               $LIB_DIR/DMET_Plus.v1.specialSNPs \
    --region-model               $LIB_DIR/DMET_Plus.v1.cn-region-models.txt \
    --probeset-model             $LIB_DIR/DMET_Plus.v1.cn-probeset-models.txt \
    --reference-input            $LIB_DIR/DMET_Plus.v1.genomic.ref.r3.a5 \
    --gt-analysis                quant-norm.sketch=50000,pm-only,brlmm-p-multi\
.CM=2.bins=100.mix=1.bic=2.lambda=0.0.HARD=3.SB=0.75.KX=0.3.KH=0.3.KXX=0.1\
.KAH=-0.1.KHB=-0.1.KYAH=-0.05.KYHB=-0.05.KYAB=-0.1.transform=MVA.AAM=2.8\
.BBM=-2.8.AAV=0.10.BBV=0.10.ABV=0.10.V=1.AAY=10.7.ABY=11.3.BBY=10.7\
.copyqc=0.00000.wobble=0.0.MS=0.001.copytype=-1.clustertype=2.CSepPen=0.5\
.ocean=0.000000000000001.cc-alleles=6.cc-type=UCHAR.cc-version=1.0 \
    --geno-call-thresh           0.001 \
    --cc-chp-output              true \
    --probeset-ids               $LIB_DIR/DMET_Plus.v1.genomic.gt.ps \
    --cn-region-gt-probeset-file $LIB_DIR/DMET_Plus.v1.cn-gt.ps \
    --chip-type                  DMET_Plus \
</pre></div><p>The three changes from the previous call include the following:</p>
<ul>
<li>The input reference file is DMET_Plus.v1.genomic.ref.r3.a5.</li>
</ul>
<ul>
<li>The '--geno-call-thresh' option value has been reduced from the default 0.1 to 0.001, representing a higher level of confidence to make the genotype calls.</li>
</ul>
<ul>
<li>The '--gt-analysis' option analysis string has had the MS and ocean parameters changed to 0.001 and 0.000000000000001, respectively.</li>
</ul>
<h2><a class="anchor" id="dyn"></a>
Dynamic boundary analysis</h2>
<p>It is possible that either the target preparation or the experimental conditions for a set of samples may differ sufficiently from those used for the training data of DMET Plus that dynamic-boundary analysis (i.e. adaptation of the cluster boundaries to new locations suggested by the empirical data) may improve the results. In cases where it provides an improvement, the usual effect is to shift cluster boundaries and increase the call rate; the accuracy typically remains about the same.</p>
<p>It is most appropriate to cluster samples in batches that correspond to equivalent processing conditions, that is, samples that share the same systematic cluster shifts. Trying to adapt to the more random spread associated with different assay conditions tends to expand the cluster variance rather than shifting the cluster. For example, if you had two batches of samples each of which had a different degree and type of cluster shifting, it would be best to perform two clustering runs, one for each batch separately, rather than a single run on all the samples.</p>
<p>In an effort to reduce the possible loss of accuracy associated with the dynamic-boundary analysis relative to the fixed-boundary analysis, the reference file and some algorithm parameters have some changes relative to the fixed boundary equivalents. The following call yields the same results as the legacy "Dynamic Genotype Boundaries" analysis configuration for DMET Console.</p>
<div class="fragment"><pre class="fragment">
  apt-dmet-genotype \
    --sample-type                genomic \
    --out-dir                    output \
    --cdf-file                   $LIB_DIR/DMET_Plus.v1.cdf \
    --cel-files                  cel_files.txt \
    --run-cn-engine              true \
    --chrX-probes                $LIB_DIR/DMET_Plus.v1.chrXprobes \
    --chrY-probes                $LIB_DIR/DMET_Plus.v1.chrYprobes \
    --special-snps               $LIB_DIR/DMET_Plus.v1.specialSNPs \
    --region-model               $LIB_DIR/DMET_Plus.v1.cn-region-models.txt \
    --probeset-model             $LIB_DIR/DMET_Plus.v1.cn-probeset-models.txt \
    --batch-name                 genomic_test_20110404 \
    --batch-info                 true \
    --reference-output           refOutput.txt \
    --reference-input            $LIB_DIR/DMET_Plus.v1.genomic.ref.r2.a5 \
    --gt-analysis                quant-norm.sketch=50000,pm-only,brlmm-p-multi\
.CM=1.bins=100.mix=1.bic=2.lambda=0.0.HARD=3.SB=0.75.KX=0.3.KH=0.3.KXX=0.1\
.KAH=-0.1.KHB=-0.1.KYAH=-0.05.KYHB=-0.05.KYAB=-0.1.transform=MVA.AAM=2.8\
.BBM=-2.8.AAV=0.10.BBV=0.10.ABV=0.10.V=1.AAY=10.7.ABY=11.3.BBY=10.7\
.copyqc=0.00000.wobble=0.0.MS=0.1.copytype=-1.clustertype=2.CSepPen=0.5\
.ocean=0.0000001.cc-alleles=6.cc-type=UCHAR.cc-version=1.0 \
    --geno-call-thresh           0.1 \
    --cc-chp-output              true \
    --probeset-ids               $LIB_DIR/DMET_Plus.v1.genomic.gt.ps \
    --cn-region-gt-probeset-file $LIB_DIR/DMET_Plus.v1.cn-gt.ps \
    --chip-type                  DMET_Plus \
</pre></div><p>The call is identical to the original fixed-boundary analysis except for four settings:</p>
<ul>
<li>The '--reference-output' option activates dynamic-boundary mode and specifies the name of the file into which the newly generated posteriors should be placed. Currently, the reference output cannot serve as reference input for a subsequent run.</li>
</ul>
<ul>
<li>The '--batch-name' option provides a name for the dynamic-boundary analysis. The choice does not affect the analysis; choose something useful for your recordkeeping.</li>
</ul>
<ul>
<li>The '--batch-info' option, when set to true, adds information to the CHP file headers that tracks the files used during the dynamic clustering.</li>
</ul>
<ul>
<li>The '--reference-input' option refers to the input reference file DMET_Plus.v1.genomic.ref.r2.a5.</li>
</ul>
<p>To process plasmids, set --sample-type to plasmid, --reference-input to DMET_Plus.v1.plasmid.ref.r2.a5, and --probeset-ids to DMET_Plus.v1.plasmid.gt.ps, respectively.</p>
<p>The following call yields the same results as the "Dynamic Genotype Boundaries &mdash; version 2" analysis configuration for DMET Console.</p>
<div class="fragment"><pre class="fragment">
  apt-dmet-genotype \
    --sample-type                genomic \
    --out-dir                    output \
    --cdf-file                   $LIB_DIR/DMET_Plus.v1.cdf \
    --cel-files                  cel_files.txt \
    --run-cn-engine              true \
    --chrX-probes                $LIB_DIR/DMET_Plus.v1.chrXprobes \
    --chrY-probes                $LIB_DIR/DMET_Plus.v1.chrYprobes \
    --special-snps               $LIB_DIR/DMET_Plus.v1.specialSNPs \
    --region-model               $LIB_DIR/DMET_Plus.v1.cn-region-models.txt \
    --probeset-model             $LIB_DIR/DMET_Plus.v1.cn-probeset-models.txt \
    --batch-name                 genomic_test_20110404 \
    --batch-info                 true \
    --reference-output           refOutput.txt \
    --reference-input            $LIB_DIR/DMET_Plus.v1.genomic.ref.r4.a5 \
    --gt-analysis                quant-norm.sketch=50000,pm-only,brlmm-p-multi\
.CM=1.bins=100.mix=1.bic=2.lambda=0.0.HARD=3.SB=0.75.KX=0.3.KH=0.3.KXX=0.1\
.KAH=-0.1.KHB=-0.1.KYAH=-0.05.KYHB=-0.05.KYAB=-0.1.transform=MVA.AAM=2.8\
.BBM=-2.8.AAV=0.10.BBV=0.10.ABV=0.10.V=1.AAY=10.7.ABY=11.3.BBY=10.7\
.copyqc=0.00000.wobble=0.0.MS=0.001.copytype=-1.clustertype=2.CSepPen=0.5\
.ocean=0.00000000001.cc-alleles=6.cc-type=UCHAR.cc-version=1.0 \
    --geno-call-thresh           0.001 \
    --cc-chp-output              true \
    --probeset-ids               $LIB_DIR/DMET_Plus.v1.genomic.gt.ps \
    --cn-region-gt-probeset-file $LIB_DIR/DMET_Plus.v1.cn-gt.ps \
    --chip-type                  DMET_Plus \
</pre></div><p>The three changes from the previous call include the following:</p>
<ul>
<li>The input reference file is DMET_Plus.v1.genomic.ref.r4.a5.</li>
</ul>
<ul>
<li>The '--geno-call-thresh' option value has been reduced from the default 0.1 to 0.001, representing a higher level of confidence to make the genotype calls.</li>
</ul>
<ul>
<li>The '--gt-analysis' option analysis string has had the MS and ocean parameters changed to 0.001 and 0.00000000001, respectively.</li>
</ul>
<h2><a class="anchor" id="access"></a>
Accessing the genotyping and copy number results</h2>
<p>Having run apt-dmet-genotype, you may wish to feed the resulting copy number and genotyping results into your bioinformatic pipeline directly. This connection to your pipeline may be facilitated through the generation of text-based output that simplifies the parsing of the output. Adding the following sequence to the end of the options for apt-dmet-genotype will generate text-formatted output files:</p>
<div class="fragment"><pre class="fragment">
  -- --summaries=true --feat-effects -- --text-output -- --table-output --feat-effects --summaries
</pre></div><p>The output directory created by apt-dmet-genotype contains four sub-directories:</p>
<ul>
<li>adc &mdash; Copy number analysis results. </li>
<li>apg &mdash; Genotyping analysis results. </li>
<li>aps &mdash; Probeset summarization for the copy number analysis. </li>
<li>chp &mdash; Binary files that contain the per-sample genotyping and copy number results (for possible input into DMET Console).</li>
</ul>
<p>The adc directory includes the dmet.copynumber.txt file, which summarizes the copy number analysis based on results from specially designed copy-number probes as well as from genotyping markers. The apg directory includes the dmet.calls.txt, dmet.forced-calls.txt, and dmet.confidences.txt files. The dmet.calls.txt file contains the genotyping calls, and the dmet.confidences.txt file contains the associated confidences. When the confidence value exceeds a threshold (controlled by the MS parameter in the --gt-analysis algorithm string above), the associated genotype is a <em>no-call</em>. The dmet.forced-calls.txt contains the genotyping calls ignoring the confidence threshold, so there is a call in every case.</p>
<p>The dmetchp.html file in the <a href="http://www.affymetrix.com/Auth/support/developer/fusion/file_formats.zip">Affymetrix Fusion SDK file format description</a> explains the various codes used in the files. Although the dmetchp.html file refers to a binary format, the same codes appear in the text output.</p>
<p>In the case of genotyping, the preceding mapping of genotyping codes results in abstract allele symbols (e.g. AA or AB) rather than the actual bases in the DNA sequence. One of the <a href="http://www.affymetrix.com/Auth/support/downloads/library_files/dmet_plus_annot_translation.zip">DMET annotation files</a>, DMET_Plus.v1.20110329.dc_annot.csv, provides a mapping from the abstract allele symbols to the DNA bases. Below are relevant fields in this file.</p>
<table  border="1" cellspacing="0">
<tr>
<th>Probe Set ID </th><th>Common Name </th><th>Design Strand </th><th>Alleles Design Strand </th><th>Allele Code </th><th>Switch Design Strand to Report </th><th>Alleles Reported Strand </th><th>Reported Strand  </th></tr>
<tr>
<td>AM_10001 </td><td>CHST3_c.*1130C&gt;G </td><td>1 </td><td>C // G </td><td>A // B </td><td>0 </td><td>C // G </td><td>+  </td></tr>
<tr>
<td>AM_10002 </td><td>CHST3_c.*1155G&gt;CorGG </td><td>1 </td><td>C // G // GG </td><td>A // B // C </td><td>0 </td><td>C // G // GG </td><td>+  </td></tr>
<tr>
<td>AM_10003 </td><td>CHST3_c.*1278C&gt;T </td><td>1 </td><td>C // T </td><td>A // B </td><td>0 </td><td>C // T </td><td>+  </td></tr>
<tr>
<td>AM_10004 </td><td>CHST3_c.*1314G&gt;A </td><td>1 </td><td>A // G </td><td>A // B </td><td>0 </td><td>A // G </td><td>+  </td></tr>
<tr>
<td>AM_10005 </td><td>CHST3_c.*1361C&gt;T </td><td>1 </td><td>C // T </td><td>A // B </td><td>0 </td><td>C // T </td><td>+  </td></tr>
<tr>
<td>AM_10006 </td><td>CHST3_c.*1844T&gt;C </td><td>-1 </td><td>A // G </td><td>A // B </td><td>1 </td><td>T // C </td><td>+  </td></tr>
</table>
<p>The order of the abstract allele code names in the <em>Allele Code</em> column is the same order as the reported allele names, represented in the <em>Alleles Reported Strand</em> column. For example, if the genotype number code for a sample at marker AM_10002 is <em>38</em> in dmet.calls.txt, a lookup in the appropriate table in the dmetchp.html file yields the abstract letter code genotype <em>C|C</em>. From the above table, this abstract genotype corresponds to <em>GG|GG</em>.</p>
<p><b></p>
<pre>Affymetrix Power Tools (APT) Release 1.18.0</pre><p></b> </p>
</div>
<hr class="footer"/><address class="footer"><small>Generated for Affymetrix Power Tools by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/></a> 1.7.1 </small></address>
</body>
</html>
